<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />
<meta content="PipeCraft2 manual. User guide for PipeCraft2" lang="en" name="description" xml:lang="en" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Pre-defined pipelines &mdash; PipeCraft 2 1.0.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Individual steps (Quick Tools)" href="quicktools.html" />
    <link rel="prev" title="General overview" href="user_guide.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> PipeCraft 2
          </a>
              <div class="version">
                1.0.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="user_guide.html">General overview</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Pre-defined pipelines</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#vsearch-otus">vsearch OTUs</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#otus-workflow-panel">OTUs workflow panel</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#dada2-asvs">DADA2 ASVs</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#asvs-workflow-panel-with-dada2">ASVs workflow panel (with DADA2)</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#quality-filtering-asvs-workflow">QUALITY FILTERING [ASVs workflow]</a></li>
<li class="toctree-l4"><a class="reference internal" href="#denoising-asvs-workflow">DENOISING [ASVs workflow]</a></li>
<li class="toctree-l4"><a class="reference internal" href="#merge-pairs-asvs-workflow">MERGE PAIRS [ASVs workflow]</a></li>
<li class="toctree-l4"><a class="reference internal" href="#chimera-filtering-asvs-workflow">CHIMERA FILTERING [ASVs workflow]</a></li>
<li class="toctree-l4"><a class="reference internal" href="#filter-asv-table-asvs-workflow">filter ASV table [ASVs workflow]</a></li>
<li class="toctree-l4"><a class="reference internal" href="#assign-taxonomy-asvs-workflow">ASSIGN TAXONOMY [ASVs workflow]</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#unoise-asvs">UNOISE ASVs</a></li>
<li class="toctree-l2"><a class="reference internal" href="#nextits">NextITS</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="quicktools.html">Individual steps (Quick Tools)</a></li>
<li class="toctree-l1"><a class="reference internal" href="postprocessing.html">Post-processing tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorial.html">Example data analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="troubleshoot.html">Troubleshooting</a></li>
<li class="toctree-l1"><a class="reference internal" href="licence.html">Licence</a></li>
<li class="toctree-l1"><a class="reference internal" href="contact.html">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="citation.html">How to cite</a></li>
<li class="toctree-l1"><a class="reference internal" href="releases.html">Releases</a></li>
<li class="toctree-l1"><a class="reference internal" href="docker_images.html">Docker images</a></li>
<li class="toctree-l1"><a class="reference internal" href="for_developers.html">For Developers</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">PipeCraft 2</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Pre-defined pipelines</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/pre-defined_pipelines.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <dl class="simple">
<dt><a class="reference internal" href="_images/PipeCraft2_icon_v2.png"><img alt="Alternative text" src="_images/PipeCraft2_icon_v2.png" style="width: 100px;" /></a></dt><dd><p><a class="reference external" href="https://github.com/pipecraft2/pipecraft">github</a></p>
</dd>
</dl>
<style> .red {color:#ff0000; font-weight:bold; font-size:16px} </style><section id="pre-defined-pipelines">
<h1>Pre-defined pipelines<a class="headerlink" href="#pre-defined-pipelines" title="Permalink to this headline"></a></h1>
<section id="vsearch-otus">
<span id="predefinedpipelines"></span><h2>vsearch OTUs<a class="headerlink" href="#vsearch-otus" title="Permalink to this headline"></a></h2>
<section id="otus-workflow-panel">
<span id="otupipe"></span><h3>OTUs workflow panel<a class="headerlink" href="#otus-workflow-panel" title="Permalink to this headline"></a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This OTU workflow works with paired-end (e.g. Illumina, MGI-Tech) as well as single-end reads (e.g. PacBio, assembled Illumina reads)</p>
</div>
<dl>
<dt>This automated workflow is mostly based on <a class="reference external" href="https://github.com/torognes/vsearch">vsearch</a> (<a class="reference external" href="https://peerj.com/articles/2584/">Rognes et. al 2016</a>) [<a class="reference external" href="_static/vsearch_manual_2.22.1.pdf">manual</a>]</dt><dd><div class="line-block">
<div class="line">Note that <code class="docutils literal notranslate"><span class="pre">demultiplexing</span></code>, <code class="docutils literal notranslate"><span class="pre">reorient</span></code> and <code class="docutils literal notranslate"><span class="pre">remove</span> <span class="pre">primers</span></code> steps are optional. Nevertheless, it is advisable to <a class="reference internal" href="quicktools.html#reorient"><span class="std std-ref">reorient</span></a> your reads (to 5’-3’) and <a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">remove primers</span></a> before proceeding.</div>
</div>
</dd>
</dl>
<div class="line-block" id="otupipe-defaults">
<div class="line"><strong>Default options:</strong></div>
<div class="line"><em>click on analyses step for more info</em></div>
</div>
<table class="docutils align-default">
<colgroup>
<col style="width: 64%" />
<col style="width: 36%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Analyses step</p></th>
<th class="head"><p>Default setting</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference internal" href="quicktools.html#demux"><span class="std std-ref">DEMULTIPLEX</span></a> (optional)</p></td>
<td><p>–</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="quicktools.html#reorient"><span class="std std-ref">REORIENT</span></a> (optional)</p></td>
<td><p>–</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">REMOVE PRIMERS</span></a> (optional)</p></td>
<td><p>–</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="quicktools.html#merge-vsearch"><span class="std std-ref">MERGE READS</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">read_R1</span></code> = \.R1</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">min_overlap</span></code> = 12</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">min_length</span></code> = 32</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">allow_merge_stagger</span></code> = TRUE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">include</span> <span class="pre">only</span> <span class="pre">R1</span></code> = FALSE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">max_diffs</span></code> = 20</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">max_Ns</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">max_len</span></code> = 600</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">keep_disjoined</span></code> = FALSE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">fastq_qmax</span></code> = 41</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="quicktools.html#qfilt-vsearch"><span class="std std-ref">QUALITY FILTERING with vsearch</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxEE</span></code> = 1</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxN</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">minLen</span></code> = 32</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">max_length</span></code> = undefined</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">qmax</span></code> = 41</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">qmin</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxee_rate</span></code> = undefined</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="quicktools.html#chimfilt-vsearch"><span class="std std-ref">CHIMERA FILTERING with uchime_denovo</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">pre_cluster</span></code> = 0.98</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">min_unique_size</span></code> = 1</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">denovo</span></code> = TRUE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">reference_based</span></code> = undefined</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">abundance_skew</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">min_h</span></code> = 0.28</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="quicktools.html#itsextractor"><span class="std std-ref">ITS Extractor</span></a> (optional)</p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">organisms</span></code> = all</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">regions</span></code> = all</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">partial</span></code> = 50</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">region_for_clustering</span></code> = ITS2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">cluster_full_and_partial</span></code> = TRUE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">e_value</span></code> = 1e-2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">scores</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">domains</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">complement</span></code> = TRUE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">only_full</span></code> = FALSE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">truncate</span></code> = TRUE</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="quicktools.html#clustering-vsearch"><span class="std std-ref">CLUSTERING with vsearch</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">OTU_type</span></code> = centroid</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">similarity_threshold</span></code> = 0.97</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">strands</span></code> = both</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">remove_singletons</span></code> = false</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">similarity_type</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">sequence_sorting</span></code> = cluster_size</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">centroid_type</span></code> = similarity</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">max_hits</span></code> = 1</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">mask</span></code> = dust</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">dbmask</span></code> = dust</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="quicktools.html#assign-taxonomy-blast"><span class="std std-ref">ASSIGN TAXONOMY with BLAST</span></a> (optional)</p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">database_file</span></code> = select a database</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">task</span></code> = blastn</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">strands</span></code> = both</div>
</div>
</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="dada2-asvs">
<h2>DADA2 ASVs<a class="headerlink" href="#dada2-asvs" title="Permalink to this headline"></a></h2>
<section id="asvs-workflow-panel-with-dada2">
<span id="asvpipe"></span><h3>ASVs workflow panel (with <a class="reference external" href="https://benjjneb.github.io/dada2/index.html">DADA2</a>)<a class="headerlink" href="#asvs-workflow-panel-with-dada2" title="Permalink to this headline"></a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Working directory must contain <strong>at least 2 samples</strong> for DADA2 pipeline.</p>
</div>
<dl>
<dt>This automated workflow is based on the <a class="reference external" href="https://benjjneb.github.io/dada2/tutorial.html">DADA2 tutorial</a></dt><dd><div class="line-block">
<div class="line">Note that <code class="docutils literal notranslate"><span class="pre">demultiplexing</span></code>, <code class="docutils literal notranslate"><span class="pre">reorienting</span></code>, and <code class="docutils literal notranslate"><span class="pre">primer</span> <span class="pre">removal</span></code> steps are optional and do not represent parts from the DADA2 tutorial. Nevertheless, it is advisable to <a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">remove primers</span></a> before proceeding with ASV generation with DADA2.</div>
</div>
</dd>
</dl>
<div class="line-block">
<div class="line">The official DADA2 manual is available <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">here</a></div>
</div>
<p id="dada2-defaults"><strong>Default options:</strong></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 60%" />
<col style="width: 40%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Analyses step</p></th>
<th class="head"><p>Default setting</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference internal" href="quicktools.html#demux"><span class="std std-ref">DEMULTIPLEX</span></a> (optional)</p></td>
<td><div class="line-block">
<div class="line">–</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="quicktools.html#reorient"><span class="std std-ref">REORIENT</span></a> (optional)</p></td>
<td><div class="line-block">
<div class="line">–</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">REMOVE PRIMERS</span></a> (optional)</p></td>
<td><div class="line-block">
<div class="line">–</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#dada2-qual-filt"><span class="std std-ref">QUALITY FILTERING</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">read_R1</span></code> = \.R1</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">read_R2</span></code> = \.R2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxEE</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxN</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">minLen</span></code> = 20</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">truncQ</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">truncLen</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxLen</span></code> = 9999</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">minQ</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">matchIDs</span></code> = TRUE</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#dada2-denoise"><span class="std std-ref">DENOISE</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">pool</span></code> = FALSE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">selfConsist</span></code> = FASLE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">qualityType</span></code> = Auto</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#dada2-merge-pairs"><span class="std std-ref">MERGE PAIRED-END READS</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">minOverlap</span></code> = 12</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxMismatch</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">trimOverhang</span></code> = FALSE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">justConcatenate</span></code> = FALSE</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#dada2-chimeras"><span class="std std-ref">CHIMERA FILTERING</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">method</span></code> = consensus</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#dada2-table-filtering"><span class="std std-ref">Filter ASV table</span></a> (optional)</p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">collapseNoMismatch</span></code> = TRUE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">by_length</span></code> = 250</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">minOverlap</span></code> = 20</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">vec</span></code> = TRUE</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#dada2-taxonomy"><span class="std std-ref">ASSIGN TAXONOMY</span></a> (optional)</p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">minBoot</span></code> = 50</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">tryRC</span></code> = FALSE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">dada2</span> <span class="pre">database</span></code> = select a database</div>
</div>
</td>
</tr>
</tbody>
</table>
<hr class="docutils" />
<section id="quality-filtering-asvs-workflow">
<span id="dada2-qual-filt"></span><h4>QUALITY FILTERING [ASVs workflow]<a class="headerlink" href="#quality-filtering-asvs-workflow" title="Permalink to this headline"></a></h4>
<p>DADA2 <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">filterAndTrim</a> function performs quality filtering on input FASTQ files based on user-selected criteria. Outputs include filtered FASTQ files located in the <code class="docutils literal notranslate"><span class="pre">qualFiltered_out</span></code> directory.</p>
<p>Quality profiles may be examined using the <a class="reference internal" href="user_guide.html#interface"><span class="std std-ref">QualityCheck module</span></a>.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 27%" />
<col style="width: 73%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">read_R1</span></code></p></td>
<td><div class="line-block">
<div class="line">applies only for <strong>paired-end</strong> data.</div>
<div class="line">Identifyer string that is common for all R1 reads</div>
<div class="line">(e.g. when all R1 files have ‘.R1’ string, then enter ‘\.R1’.</div>
<div class="line">Note that backslash is only needed to escape dot regex; e.g.</div>
<div class="line">when all R1 files have ‘_R1’ string, then enter ‘_R1’.).</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">read_R2</span></code></p></td>
<td><div class="line-block">
<div class="line">applies only for <strong>paired-end</strong> data.</div>
<div class="line">Identifyer string that is common for all R2 reads</div>
<div class="line">(e.g. when all R2 files have ‘.R2’ string, then enter ‘\.R2’.</div>
<div class="line">Note that backslash is only needed to escape dot regex; e.g.</div>
<div class="line">when all R2 files have ‘_R1’ string, then enter ‘_R2’.).</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">maxEE</span></code></p></td>
<td><div class="line-block">
<div class="line">discard sequences with more than the specified number of expected errors</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">maxN</span></code></p></td>
<td><div class="line-block">
<div class="line">discard sequences with more than the specified number of N’s (ambiguous bases)</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">minLen</span></code></p></td>
<td><div class="line-block">
<div class="line">remove reads with length less than minLen. minLen is enforced</div>
<div class="line">after all other trimming and truncation</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">truncQ</span></code></p></td>
<td><div class="line-block">
<div class="line">truncate reads at the first instance of a quality score less than or equal to truncQ</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">truncLen</span></code></p></td>
<td><div class="line-block">
<div class="line">truncate reads after truncLen bases</div>
<div class="line">(applies to <strong>R1 reads</strong> when working with <strong>paired-end</strong> data).</div>
<div class="line">Reads shorter than this are discarded.</div>
<div class="line">Explore quality profiles (with QualityCheck module) and</div>
<div class="line">see whether poor quality ends needs to be truncated</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">truncLen_R2</span></code></p></td>
<td><div class="line-block">
<div class="line">applies only for <strong>paired-end</strong> data.</div>
<div class="line">Truncate <strong>R2 reads</strong> after truncLen bases.</div>
<div class="line">Reads shorter than this are discarded.</div>
<div class="line">Explore quality profiles (with QualityCheck module) and</div>
<div class="line">see whether poor quality ends needs to truncated</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">maxLen</span></code></p></td>
<td><div class="line-block">
<div class="line">remove reads with length greater than maxLen.</div>
<div class="line">maxLen is enforced on the raw reads.</div>
<div class="line">In dada2, the default = Inf, but here set as 9999</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">minQ</span></code></p></td>
<td><div class="line-block">
<div class="line">after truncation, reads contain a quality score below minQ will be discarded</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">matchIDs</span></code></p></td>
<td><div class="line-block">
<div class="line">applies only for <strong>paired-end</strong> data.</div>
<div class="line">If TRUE, then double-checking (with seqkit pair) that only paired reads</div>
<div class="line">that share ids are outputted.</div>
<div class="line"><span class="red">Note that ‘seqkit’ will be used for this process</span>, because when</div>
<div class="line">using e.g. SRA fastq files where original fastq headers have been</div>
<div class="line">replaced, dada2 does not recognize those fastq id strings</div>
</div>
</td>
</tr>
</tbody>
</table>
<p>see <a class="reference internal" href="#dada2-defaults"><span class="std std-ref">default settings</span></a></p>
<hr class="docutils" />
</section>
<section id="denoising-asvs-workflow">
<span id="dada2-denoise"></span><h4>DENOISING [ASVs workflow]<a class="headerlink" href="#denoising-asvs-workflow" title="Permalink to this headline"></a></h4>
<p>DADA2 <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">dada</a> function to remove sequencing errors.
Outputs filtered fasta files into <code class="docutils literal notranslate"><span class="pre">denoised_assembled.dada2</span></code> directory.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 18%" />
<col style="width: 82%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">pool</span></code></p></td>
<td><div class="line-block">
<div class="line">if TRUE, the algorithm will pool together all samples prior to sample inference.</div>
<div class="line">Pooling improves the detection of rare variants, but is computationally more expensive.</div>
<div class="line">If pool = ‘pseudo’, the algorithm will perform pseudo-pooling between individually</div>
<div class="line">processed samples.</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">selfConsist</span></code></p></td>
<td><div class="line-block">
<div class="line">if TRUE, the algorithm will alternate between sample inference and error rate estimation</div>
<div class="line">until convergence</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">qualityType</span></code></p></td>
<td><div class="line-block">
<div class="line">‘Auto’ means to attempt to auto-detect the fastq quality encoding.</div>
<div class="line">This may fail for PacBio files with uniformly high quality scores,</div>
<div class="line">in which case use ‘FastqQuality’</div>
</div>
</td>
</tr>
</tbody>
</table>
<p>see <a class="reference internal" href="#dada2-defaults"><span class="std std-ref">default settings</span></a></p>
<hr class="docutils" />
</section>
<section id="merge-pairs-asvs-workflow">
<span id="dada2-merge-pairs"></span><h4>MERGE PAIRS [ASVs workflow]<a class="headerlink" href="#merge-pairs-asvs-workflow" title="Permalink to this headline"></a></h4>
<p>DADA2 <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">mergePairs</a> function to merge paired-end reads.
Outputs merged fasta files into <code class="docutils literal notranslate"><span class="pre">denoised_assembled.dada2</span></code> directory.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 19%" />
<col style="width: 81%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">minOverlap</span></code></p></td>
<td><div class="line-block">
<div class="line">the minimum length of the overlap required for merging the forward and reverse reads</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">maxMismatch</span></code></p></td>
<td><div class="line-block">
<div class="line">the maximum mismatches allowed in the overlap region</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">trimOverhang</span></code></p></td>
<td><div class="line-block">
<div class="line">if TRUE, overhangs in the alignment between the forwards and reverse read are</div>
<div class="line">trimmed off. Overhangs are when the reverse read extends past the start of</div>
<div class="line">the forward read, and vice-versa, as can happen when reads are longer than the</div>
<div class="line">amplicon and read into the other-direction primer region</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">justConcatenate</span></code></p></td>
<td><div class="line-block">
<div class="line">if TRUE, the forward and reverse-complemented reverse read are concatenated</div>
<div class="line">rather than merged, with a NNNNNNNNNN (10 Ns) spacer inserted between them</div>
</div>
</td>
</tr>
</tbody>
</table>
<p>see <a class="reference internal" href="#dada2-defaults"><span class="std std-ref">default settings</span></a></p>
</section>
<hr class="docutils" id="dada2-chimeras" />
<section id="chimera-filtering-asvs-workflow">
<h4>CHIMERA FILTERING [ASVs workflow]<a class="headerlink" href="#chimera-filtering-asvs-workflow" title="Permalink to this headline"></a></h4>
<p>DADA2 <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">removeBimeraDenovo</a> function to remove chimeras.
Outputs filtered fasta files into <code class="docutils literal notranslate"><span class="pre">chimeraFiltered_out.dada2</span></code> and final ASVs to <code class="docutils literal notranslate"><span class="pre">ASVs_out.dada2</span></code> directory.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 19%" />
<col style="width: 81%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">method</span></code></p></td>
<td><div class="line-block">
<div class="line">‘consensus’ - the samples are independently checked for chimeras, and a consensus</div>
<div class="line">decision on each sequence variant is made.</div>
<div class="line">If ‘pooled’, the samples are all pooled together for chimera identification.</div>
<div class="line">If ‘per-sample’, the samples are independently checked for chimeras</div>
</div>
</td>
</tr>
</tbody>
</table>
<p>see <a class="reference internal" href="#dada2-defaults"><span class="std std-ref">default settings</span></a></p>
</section>
<hr class="docutils" id="dada2-table-filtering" />
<section id="filter-asv-table-asvs-workflow">
<h4>filter ASV table [ASVs workflow]<a class="headerlink" href="#filter-asv-table-asvs-workflow" title="Permalink to this headline"></a></h4>
<p>DADA2 <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">collapseNoMismatch</a> function to collapse identical ASVs;
and ASVs filtering based on minimum accepted sequence length (custom R functions).
Outputs filtered ASV table and fasta files into <code class="docutils literal notranslate"><span class="pre">ASVs_out.dada2/filtered</span></code> directory.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 27%" />
<col style="width: 73%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">collapseNoMismatch</span></code></p></td>
<td><div class="line-block">
<div class="line">collapses ASVs that are identical up to shifts or</div>
<div class="line">length variation, i.e. that have no mismatches or internal indels</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">by_length</span></code></p></td>
<td><div class="line-block">
<div class="line">discard ASVs from the ASV table that are shorter than specified</div>
<div class="line">value (in base pairs). Value 0 means OFF, no filtering by length</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">minOverlap</span></code></p></td>
<td><div class="line-block">
<div class="line">collapseNoMismatch setting. Default = 20. The minimum overlap of</div>
<div class="line">base pairs between ASV sequences required to collapse them together</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">vec</span></code></p></td>
<td><div class="line-block">
<div class="line">collapseNoMismatch setting. Default = TRUE. Use the vectorized</div>
<div class="line">aligner. Should be turned off if sequences exceed 2kb in length</div>
</div>
</td>
</tr>
</tbody>
</table>
<p>see <a class="reference internal" href="#dada2-defaults"><span class="std std-ref">default settings</span></a></p>
<hr class="docutils" />
</section>
<section id="assign-taxonomy-asvs-workflow">
<span id="dada2-taxonomy"></span><h4>ASSIGN TAXONOMY [ASVs workflow]<a class="headerlink" href="#assign-taxonomy-asvs-workflow" title="Permalink to this headline"></a></h4>
<p>DADA2 <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">assignTaxonomy</a> function to classify ASVs.
Outputs classified fasta files into <code class="docutils literal notranslate"><span class="pre">taxonomy_out.dada2</span></code> directory.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 16%" />
<col style="width: 84%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">minBoot</span></code></p></td>
<td><div class="line-block">
<div class="line">the minimum bootstrap confidence for assigning a taxonomic level</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">tryRC</span></code></p></td>
<td><div class="line-block">
<div class="line">the reverse-complement of each sequences will be used for classification</div>
<div class="line">if it is a better match to the reference sequences than the forward sequence</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">dada2</span> <span class="pre">database</span></code></p></td>
<td><div class="line-block">
<div class="line">select a reference database fasta file for taxonomy annotation</div>
<div class="line"><a class="reference external" href="https://benjjneb.github.io/dada2/training.html">Download DADA2-formatted reference databases here</a></div>
</div>
</td>
</tr>
</tbody>
</table>
<p>see <a class="reference internal" href="#dada2-defaults"><span class="std std-ref">default settings</span></a></p>
</section>
</section>
</section>
<hr class="docutils" />
<section id="unoise-asvs">
<h2>UNOISE ASVs<a class="headerlink" href="#unoise-asvs" title="Permalink to this headline"></a></h2>
<p>UNOISE3 pipeline for making ASVs (zOTUs) + optionally automatic clustering of those ASVs.
Updating this section soon.</p>
</section>
<section id="nextits">
<h2>NextITS<a class="headerlink" href="#nextits" title="Permalink to this headline"></a></h2>
<p>NextITS pipeline for PacBio ITS sequences.
Updating this section soon.</p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="user_guide.html" class="btn btn-neutral float-left" title="General overview" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="quicktools.html" class="btn btn-neutral float-right" title="Individual steps (Quick Tools)" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Sten Anslan.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>