<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />
<meta content="PipeCraft2 manual. User guide for PipeCraft2" lang="en" name="description" xml:lang="en" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Pre-defined pipelines &mdash; PipeCraft2 1.0.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=8d563738"></script>
        <script src="_static/doctools.js?v=9a2dae69"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Individual steps (Quick Tools)" href="quicktools.html" />
    <link rel="prev" title="QuickStart" href="quickstart.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            PipeCraft2
          </a>
              <div class="version">
                1.0.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="quickstart.html">QuickStart</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Pre-defined pipelines</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#vsearch-otus">vsearch OTUs</a></li>
<li class="toctree-l2"><a class="reference internal" href="#dada2-asvs">DADA2 ASVs</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#quality-filtering">QUALITY FILTERING</a></li>
<li class="toctree-l3"><a class="reference internal" href="#denoising">DENOISING</a></li>
<li class="toctree-l3"><a class="reference internal" href="#merge-pairs">MERGE PAIRS</a></li>
<li class="toctree-l3"><a class="reference internal" href="#chimera-filtering">CHIMERA FILTERING</a></li>
<li class="toctree-l3"><a class="reference internal" href="#filter-asv-table">filter ASV table</a></li>
<li class="toctree-l3"><a class="reference internal" href="#assign-taxonomy">ASSIGN TAXONOMY</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#unoise-asvs">UNOISE ASVs</a></li>
<li class="toctree-l2"><a class="reference internal" href="#nextits">NextITS</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#single-sequencing-run">Single sequencing run</a></li>
<li class="toctree-l3"><a class="reference internal" href="#multiple-sequencing-runs">Multiple sequencing runs</a></li>
<li class="toctree-l3"><a class="reference internal" href="#cut-primers">Cut primers</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id2">Quality filtering</a></li>
<li class="toctree-l3"><a class="reference internal" href="#its-extraction">ITS extraction</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id3">Chimera filtering</a></li>
<li class="toctree-l3"><a class="reference internal" href="#tag-jump-correction">Tag-jump correction</a></li>
<li class="toctree-l3"><a class="reference internal" href="#unoise-denoising">UNOISE denoising</a></li>
<li class="toctree-l3"><a class="reference internal" href="#clustering">Clustering</a></li>
<li class="toctree-l3"><a class="reference internal" href="#post-clustering-with-lulu">Post-clustering with LULU</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="quicktools.html">Individual steps (Quick Tools)</a></li>
<li class="toctree-l1"><a class="reference internal" href="postprocessing.html">Post-processing tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorial.html">Example data analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="troubleshoot.html">Troubleshooting</a></li>
<li class="toctree-l1"><a class="reference internal" href="licence.html">Licence</a></li>
<li class="toctree-l1"><a class="reference internal" href="contact.html">Contact and Acknowledgements</a></li>
<li class="toctree-l1"><a class="reference internal" href="citation.html">How to cite</a></li>
<li class="toctree-l1"><a class="reference internal" href="releases.html">Releases</a></li>
<li class="toctree-l1"><a class="reference internal" href="docker_images.html">Docker images</a></li>
<li class="toctree-l1"><a class="reference internal" href="for_developers.html">For Developers</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">PipeCraft2</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Pre-defined pipelines</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/pre-defined_pipelines.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <dl class="simple">
<dt><a class="reference internal" href="_images/PipeCraft2_icon_v2.png"><img alt="Alternative text" src="_images/PipeCraft2_icon_v2.png" style="width: 100px;" /></a></dt><dd><p><a class="reference external" href="https://github.com/pipecraft2/pipecraft">github</a></p>
</dd>
</dl>
<style> .red {color:#ff0000; font-weight:bold; font-size:16px} </style><section id="pre-defined-pipelines">
<h1>Pre-defined pipelines<a class="headerlink" href="#pre-defined-pipelines" title="Link to this heading"></a></h1>
<section id="vsearch-otus">
<span id="predefinedpipelines"></span><h2>vsearch OTUs<a class="headerlink" href="#vsearch-otus" title="Link to this heading"></a></h2>
<<<<<<< HEAD
<span class="target" id="vsearchotus"></span><p>This automated workflow is mostly based on <a class="reference external" href="https://github.com/torognes/vsearch">vsearch</a> (<a class="reference external" href="https://peerj.com/articles/2584/">Rognes et. al 2016</a>) [<a class="reference external" href="_static/vsearch_manual_2.22.1.pdf">manual</a>]
to form <strong>OTUs and an OTU table</strong>. This input is the directory that contains per-sample fastq files (<strong>demultiplexed data</strong>).</p>
<div class="line-block" id="vsearchotus-defaults">
=======
<section id="otus-workflow-panel">
<span id="otupipe"></span><h3>OTUs workflow panel<a class="headerlink" href="#otus-workflow-panel" title="Link to this heading"></a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This OTU workflow works with paired-end (e.g. Illumina, MGI-Tech) as well as single-end reads (e.g. PacBio, assembled Illumina reads)</p>
</div>
<dl>
<dt>This automated workflow is mostly based on <a class="reference external" href="https://github.com/torognes/vsearch">vsearch</a> (<a class="reference external" href="https://peerj.com/articles/2584/">Rognes et. al 2016</a>) [<a class="reference external" href="_static/vsearch_manual_2.22.1.pdf">manual</a>]</dt><dd><div class="line-block">
<div class="line">Note that <code class="docutils literal notranslate"><span class="pre">demultiplexing</span></code>, <code class="docutils literal notranslate"><span class="pre">reorient</span></code> and <code class="docutils literal notranslate"><span class="pre">remove</span> <span class="pre">primers</span></code> steps are optional. Nevertheless, it is advisable to <a class="reference internal" href="quicktools.html#reorient"><span class="std std-ref">reorient</span></a> your reads (to 5’-3’) and <a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">remove primers</span></a> before proceeding.</div>
</div>
</dd>
</dl>
<div class="line-block" id="otupipe-defaults">
>>>>>>> c197e31f5a449b4e3650a9545d50562e2de36a2e
<div class="line"><strong>Default options:</strong></div>
<div class="line"><em>click on analyses step for more info</em></div>
</div>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Analyses step</p></th>
<th class="head"><p>Default setting</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">CUT PRIMERS</span></a> (optional)</p></td>
<td><p>–</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="quicktools.html#merge-vsearch"><span class="std std-ref">MERGE READS</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">read_R1</span></code> = \.R1</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">min_overlap</span></code> = 12</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">min_length</span></code> = 32</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">allow_merge_stagger</span></code> = TRUE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">include</span> <span class="pre">only</span> <span class="pre">R1</span></code> = FALSE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">max_diffs</span></code> = 20</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">max_Ns</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">max_len</span></code> = 600</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">keep_disjoined</span></code> = FALSE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">fastq_qmax</span></code> = 41</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="quicktools.html#qfilt-vsearch"><span class="std std-ref">QUALITY FILTERING with vsearch</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxEE</span></code> = 1</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxN</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">minLen</span></code> = 32</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">max_length</span></code> = undefined</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">qmax</span></code> = 41</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">qmin</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxee_rate</span></code> = undefined</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="quicktools.html#chimfilt-vsearch"><span class="std std-ref">CHIMERA FILTERING with uchime_denovo</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">pre_cluster</span></code> = 0.98</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">min_unique_size</span></code> = 1</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">denovo</span></code> = TRUE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">reference_based</span></code> = undefined</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">abundance_skew</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">min_h</span></code> = 0.28</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="quicktools.html#itsextractor"><span class="std std-ref">ITS Extractor</span></a> (optional)</p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">organisms</span></code> = all</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">regions</span></code> = all</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">partial</span></code> = 50</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">region_for_clustering</span></code> = ITS2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">cluster_full_and_partial</span></code> = TRUE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">e_value</span></code> = 1e-2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">scores</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">domains</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">complement</span></code> = TRUE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">only_full</span></code> = FALSE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">truncate</span></code> = TRUE</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="quicktools.html#clustering-vsearch"><span class="std std-ref">CLUSTERING with vsearch</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">OTU_type</span></code> = centroid</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">similarity_threshold</span></code> = 0.97</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">strands</span></code> = both</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">remove_singletons</span></code> = false</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">similarity_type</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">sequence_sorting</span></code> = cluster_size</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">centroid_type</span></code> = similarity</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">max_hits</span></code> = 1</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">mask</span></code> = dust</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">dbmask</span></code> = dust</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="quicktools.html#assign-taxonomy-blast"><span class="std std-ref">ASSIGN TAXONOMY with BLAST</span></a> (optional)</p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">database_file</span></code> = select a database</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">task</span></code> = blastn</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">strands</span></code> = both</div>
</div>
</td>
</tr>
</tbody>
</table>
</section>
<section id="dada2-asvs">
<h2>DADA2 ASVs<a class="headerlink" href="#dada2-asvs" title="Link to this heading"></a></h2>
<<<<<<< HEAD
<p id="asvpipe">This pre-defined workflow is based on the <a class="reference external" href="https://benjjneb.github.io/dada2/tutorial.html">DADA2 tutorial</a> to form <strong>ASVs and an ASV table</strong>.
This input is the directory that contains per-sample fastq files (<strong>demultiplexed data</strong>).</p>
=======
<section id="asvs-workflow-panel-with-dada2">
<span id="asvpipe"></span><h3>ASVs workflow panel (with <a class="reference external" href="https://benjjneb.github.io/dada2/index.html">DADA2</a>)<a class="headerlink" href="#asvs-workflow-panel-with-dada2" title="Link to this heading"></a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Working directory must contain <strong>at least 2 samples</strong> for DADA2 pipeline.</p>
</div>
<dl>
<dt>This automated workflow is based on the <a class="reference external" href="https://benjjneb.github.io/dada2/tutorial.html">DADA2 tutorial</a></dt><dd><div class="line-block">
<div class="line">Note that <code class="docutils literal notranslate"><span class="pre">demultiplexing</span></code>, <code class="docutils literal notranslate"><span class="pre">reorienting</span></code>, and <code class="docutils literal notranslate"><span class="pre">primer</span> <span class="pre">removal</span></code> steps are optional and do not represent parts from the DADA2 tutorial. Nevertheless, it is advisable to <a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">remove primers</span></a> before proceeding with ASV generation with DADA2.</div>
</div>
</dd>
</dl>
>>>>>>> c197e31f5a449b4e3650a9545d50562e2de36a2e
<div class="line-block">
<div class="line">Note that <code class="docutils literal notranslate"><span class="pre">primer</span> <span class="pre">removal</span></code> step do not represent parts from the DADA2 tutorial. Nevertheless, it is advisable to <a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">remove primers</span></a> before proceeding with ASV generation with DADA2.</div>
</div>
<<<<<<< HEAD
<table class="docutils align-default" id="dada2-modes">
=======
<p id="dada2-defaults"><strong>Default options:</strong></p>
<table class="docutils align-default">
>>>>>>> c197e31f5a449b4e3650a9545d50562e2de36a2e
<thead>
<tr class="row-odd"><th class="head"><p>pipeline mode</p></th>
<th class="head"><p>when do use</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><strong>FORWARD</strong></p></td>
<td><div class="line-block">
<div class="line">for paired-end Illumina data where amplicons</div>
<div class="line">are expected to be in uniform orientation</div>
<div class="line">(e.g., all reads are in 5’-3’ orientation).</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><strong>MIXED</strong></p></td>
<td><div class="line-block">
<div class="line">for paired-end Illumina data where amplicons</div>
<div class="line">are expected to be both, in 5’-3’ (forward)</div>
<div class="line">and 3’-5’ (reverse) oriented.</div>
<div class="line">In that mode, <code class="docutils literal notranslate"><span class="pre">CUT</span> <span class="pre">PRIMERS</span></code> is mandatory,</div>
<div class="line">and generates separate directories for forward</div>
<div class="line">and reverse oriented sequences, which will pass</div>
<div class="line">DADA2 pipeline individually. After merging the paired ends,</div>
<div class="line">the reverse oriented sequences are reverse complemented</div>
<div class="line">and aggregated with the forward reads for chimera filtering</div>
<div class="line">and ASV table generation. The output ASVs are all 5’-3’ oriented.</div>
<div class="line"><strong>Here, make sure that the R1 and R2 identifiers are</strong></div>
<div class="line"><strong>\.R1 and \.R2 in the QUALITY FILTERING step</strong></div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><strong>PACBIO</strong></p></td>
<td><div class="line-block">
<div class="line">for single-end PacBio data. <code class="docutils literal notranslate"><span class="pre">CUT</span> <span class="pre">PRIMERS</span></code> step for single-end</div>
<div class="line">data will reoriente all reads to 5’-3’ (forward) orientation.</div>
</div>
</td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Working directory must contain <strong>at least 2 samples</strong> for DADA2 pipeline.</p>
</div>
<p id="dada2-defaults"><strong>Default options:</strong></p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Analyses step</p></th>
<th class="head"><p>Default setting</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">CUT PRIMERS</span></a> (optional)</p></td>
<td><div class="line-block">
<div class="line">–</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#dada2-qual-filt"><span class="std std-ref">QUALITY FILTERING</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">read_R1</span></code> = \.R1</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">read_R2</span></code> = \.R2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxEE</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxN</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">minLen</span></code> = 20</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">truncQ</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">truncLen</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxLen</span></code> = 9999</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">minQ</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">matchIDs</span></code> = TRUE</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#dada2-denoise"><span class="std std-ref">DENOISE</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">pool</span></code> = FALSE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">selfConsist</span></code> = FASLE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">qualityType</span></code> = Auto</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#dada2-merge-pairs"><span class="std std-ref">MERGE PAIRED-END READS</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">minOverlap</span></code> = 12</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">maxMismatch</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">trimOverhang</span></code> = FALSE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">justConcatenate</span></code> = FALSE</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#dada2-chimeras"><span class="std std-ref">CHIMERA FILTERING</span></a></p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">method</span></code> = consensus</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#dada2-table-filtering"><span class="std std-ref">Filter ASV table</span></a> (optional)</p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">collapseNoMismatch</span></code> = TRUE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">by_length</span></code> = 250</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">minOverlap</span></code> = 20</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">vec</span></code> = TRUE</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#dada2-taxonomy"><span class="std std-ref">ASSIGN TAXONOMY</span></a> (optional)</p></td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">minBoot</span></code> = 50</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">tryRC</span></code> = FALSE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">dada2</span> <span class="pre">database</span></code> = select a database</div>
</div>
</td>
</tr>
</tbody>
</table>
<hr class="docutils" />
<<<<<<< HEAD
<section id="quality-filtering">
<span id="dada2-qual-filt"></span><h3>QUALITY FILTERING<a class="headerlink" href="#quality-filtering" title="Link to this heading"></a></h3>
=======
<section id="quality-filtering-asvs-workflow">
<span id="dada2-qual-filt"></span><h4>QUALITY FILTERING [ASVs workflow]<a class="headerlink" href="#quality-filtering-asvs-workflow" title="Link to this heading"></a></h4>
>>>>>>> c197e31f5a449b4e3650a9545d50562e2de36a2e
<p>DADA2 <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">filterAndTrim</a> function performs quality filtering on input FASTQ files based on user-selected criteria. Outputs include filtered FASTQ files located in the <code class="docutils literal notranslate"><span class="pre">qualFiltered_out</span></code> directory.</p>
<p>Quality profiles may be examined using the <a class="reference internal" href="index.html#interface"><span class="std std-ref">QualityCheck module</span></a>.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">read_R1</span></code></p></td>
<td><div class="line-block">
<div class="line">applies only for <strong>paired-end</strong> data.</div>
<div class="line">Identifyer string that is common for all R1 reads</div>
<div class="line">(e.g. when all R1 files have ‘.R1’ string, then enter ‘\.R1’.</div>
<div class="line">Note that backslash is only needed to escape dot regex; e.g.</div>
<div class="line">when all R1 files have ‘_R1’ string, then enter ‘_R1’.).</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">read_R2</span></code></p></td>
<td><div class="line-block">
<div class="line">applies only for <strong>paired-end</strong> data.</div>
<div class="line">Identifyer string that is common for all R2 reads</div>
<div class="line">(e.g. when all R2 files have ‘.R2’ string, then enter ‘\.R2’.</div>
<div class="line">Note that backslash is only needed to escape dot regex; e.g.</div>
<div class="line">when all R2 files have ‘_R1’ string, then enter ‘_R2’.).</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">maxEE</span></code></p></td>
<td><div class="line-block">
<div class="line">discard sequences with more than the specified number of expected errors</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">maxN</span></code></p></td>
<td><div class="line-block">
<div class="line">discard sequences with more than the specified number of N’s (ambiguous bases)</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">minLen</span></code></p></td>
<td><div class="line-block">
<div class="line">remove reads with length less than minLen. minLen is enforced</div>
<div class="line">after all other trimming and truncation</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">truncQ</span></code></p></td>
<td><div class="line-block">
<div class="line">truncate reads at the first instance of a quality score less than or equal to truncQ</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">truncLen</span></code></p></td>
<td><div class="line-block">
<div class="line">truncate reads after truncLen bases</div>
<div class="line">(applies to <strong>R1 reads</strong> when working with <strong>paired-end</strong> data).</div>
<div class="line">Reads shorter than this are discarded.</div>
<div class="line">Explore quality profiles (with QualityCheck module) and</div>
<div class="line">see whether poor quality ends needs to be truncated</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">truncLen_R2</span></code></p></td>
<td><div class="line-block">
<div class="line">applies only for <strong>paired-end</strong> data.</div>
<div class="line">Truncate <strong>R2 reads</strong> after truncLen bases.</div>
<div class="line">Reads shorter than this are discarded.</div>
<div class="line">Explore quality profiles (with QualityCheck module) and</div>
<div class="line">see whether poor quality ends needs to truncated</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">maxLen</span></code></p></td>
<td><div class="line-block">
<div class="line">remove reads with length greater than maxLen.</div>
<div class="line">maxLen is enforced on the raw reads.</div>
<div class="line">In dada2, the default = Inf, but here set as 9999</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">minQ</span></code></p></td>
<td><div class="line-block">
<div class="line">after truncation, reads contain a quality score below minQ will be discarded</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">matchIDs</span></code></p></td>
<td><div class="line-block">
<div class="line">applies only for <strong>paired-end</strong> data.</div>
<div class="line">If TRUE, then double-checking (with seqkit pair) that only paired reads</div>
<div class="line">that share ids are outputted.</div>
<div class="line"><span class="red">Note that ‘seqkit’ will be used for this process</span>, because when</div>
<div class="line">using e.g. SRA fastq files where original fastq headers have been</div>
<div class="line">replaced, dada2 does not recognize those fastq id strings</div>
</div>
</td>
</tr>
</tbody>
</table>
<p>see <a class="reference internal" href="#dada2-defaults"><span class="std std-ref">default settings</span></a></p>
<hr class="docutils" />
</section>
<<<<<<< HEAD
<section id="denoising">
<span id="dada2-denoise"></span><h3>DENOISING<a class="headerlink" href="#denoising" title="Link to this heading"></a></h3>
=======
<section id="denoising-asvs-workflow">
<span id="dada2-denoise"></span><h4>DENOISING [ASVs workflow]<a class="headerlink" href="#denoising-asvs-workflow" title="Link to this heading"></a></h4>
>>>>>>> c197e31f5a449b4e3650a9545d50562e2de36a2e
<p>DADA2 <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">dada</a> function to remove sequencing errors.
Outputs filtered fasta files into <code class="docutils literal notranslate"><span class="pre">denoised_assembled.dada2</span></code> directory.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">pool</span></code></p></td>
<td><div class="line-block">
<div class="line">if TRUE, the algorithm will pool together all samples prior to sample inference.</div>
<div class="line">Pooling improves the detection of rare variants, but is computationally more expensive.</div>
<div class="line">If pool = ‘pseudo’, the algorithm will perform pseudo-pooling between individually</div>
<div class="line">processed samples.</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">selfConsist</span></code></p></td>
<td><div class="line-block">
<div class="line">if TRUE, the algorithm will alternate between sample inference and error rate estimation</div>
<div class="line">until convergence</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">qualityType</span></code></p></td>
<td><div class="line-block">
<div class="line">‘Auto’ means to attempt to auto-detect the fastq quality encoding.</div>
<div class="line">This may fail for PacBio files with uniformly high quality scores,</div>
<div class="line">in which case use ‘FastqQuality’</div>
</div>
</td>
</tr>
</tbody>
</table>
<p>see <a class="reference internal" href="#dada2-defaults"><span class="std std-ref">default settings</span></a></p>
<hr class="docutils" />
</section>
<<<<<<< HEAD
<section id="merge-pairs">
<span id="dada2-merge-pairs"></span><h3>MERGE PAIRS<a class="headerlink" href="#merge-pairs" title="Link to this heading"></a></h3>
=======
<section id="merge-pairs-asvs-workflow">
<span id="dada2-merge-pairs"></span><h4>MERGE PAIRS [ASVs workflow]<a class="headerlink" href="#merge-pairs-asvs-workflow" title="Link to this heading"></a></h4>
>>>>>>> c197e31f5a449b4e3650a9545d50562e2de36a2e
<p>DADA2 <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">mergePairs</a> function to merge paired-end reads.
Outputs merged fasta files into <code class="docutils literal notranslate"><span class="pre">denoised_assembled.dada2</span></code> directory.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">minOverlap</span></code></p></td>
<td><div class="line-block">
<div class="line">the minimum length of the overlap required for merging the forward and reverse reads</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">maxMismatch</span></code></p></td>
<td><div class="line-block">
<div class="line">the maximum mismatches allowed in the overlap region</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">trimOverhang</span></code></p></td>
<td><div class="line-block">
<div class="line">if TRUE, overhangs in the alignment between the forwards and reverse read are</div>
<div class="line">trimmed off. Overhangs are when the reverse read extends past the start of</div>
<div class="line">the forward read, and vice-versa, as can happen when reads are longer than the</div>
<div class="line">amplicon and read into the other-direction primer region</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">justConcatenate</span></code></p></td>
<td><div class="line-block">
<div class="line">if TRUE, the forward and reverse-complemented reverse read are concatenated</div>
<div class="line">rather than merged, with a NNNNNNNNNN (10 Ns) spacer inserted between them</div>
</div>
</td>
</tr>
</tbody>
</table>
<p>see <a class="reference internal" href="#dada2-defaults"><span class="std std-ref">default settings</span></a></p>
</section>
<hr class="docutils" id="dada2-chimeras" />
<<<<<<< HEAD
<section id="chimera-filtering">
<h3>CHIMERA FILTERING<a class="headerlink" href="#chimera-filtering" title="Link to this heading"></a></h3>
=======
<section id="chimera-filtering-asvs-workflow">
<h4>CHIMERA FILTERING [ASVs workflow]<a class="headerlink" href="#chimera-filtering-asvs-workflow" title="Link to this heading"></a></h4>
>>>>>>> c197e31f5a449b4e3650a9545d50562e2de36a2e
<p>DADA2 <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">removeBimeraDenovo</a> function to remove chimeras.
Outputs filtered fasta files into <code class="docutils literal notranslate"><span class="pre">chimeraFiltered_out.dada2</span></code> and final ASVs to <code class="docutils literal notranslate"><span class="pre">ASVs_out.dada2</span></code> directory.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">method</span></code></p></td>
<td><div class="line-block">
<div class="line">‘consensus’ - the samples are independently checked for chimeras, and a consensus</div>
<div class="line">decision on each sequence variant is made.</div>
<div class="line">If ‘pooled’, the samples are all pooled together for chimera identification.</div>
<div class="line">If ‘per-sample’, the samples are independently checked for chimeras</div>
</div>
</td>
</tr>
</tbody>
</table>
<p>see <a class="reference internal" href="#dada2-defaults"><span class="std std-ref">default settings</span></a></p>
</section>
<hr class="docutils" id="dada2-table-filtering" />
<<<<<<< HEAD
<section id="filter-asv-table">
<h3>filter ASV table<a class="headerlink" href="#filter-asv-table" title="Link to this heading"></a></h3>
=======
<section id="filter-asv-table-asvs-workflow">
<h4>filter ASV table [ASVs workflow]<a class="headerlink" href="#filter-asv-table-asvs-workflow" title="Link to this heading"></a></h4>
>>>>>>> c197e31f5a449b4e3650a9545d50562e2de36a2e
<p>DADA2 <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">collapseNoMismatch</a> function to collapse identical ASVs;
and ASVs filtering based on minimum accepted sequence length (custom R functions).
Outputs filtered ASV table and fasta files into <code class="docutils literal notranslate"><span class="pre">ASVs_out.dada2/filtered</span></code> directory.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">collapseNoMismatch</span></code></p></td>
<td><div class="line-block">
<div class="line">collapses ASVs that are identical up to shifts or</div>
<div class="line">length variation, i.e. that have no mismatches or internal indels</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">by_length</span></code></p></td>
<td><div class="line-block">
<div class="line">discard ASVs from the ASV table that are shorter than specified</div>
<div class="line">value (in base pairs). Value 0 means OFF, no filtering by length</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">minOverlap</span></code></p></td>
<td><div class="line-block">
<div class="line">collapseNoMismatch setting. Default = 20. The minimum overlap of</div>
<div class="line">base pairs between ASV sequences required to collapse them together</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">vec</span></code></p></td>
<td><div class="line-block">
<div class="line">collapseNoMismatch setting. Default = TRUE. Use the vectorized</div>
<div class="line">aligner. Should be turned off if sequences exceed 2kb in length</div>
</div>
</td>
</tr>
</tbody>
</table>
<p>see <a class="reference internal" href="#dada2-defaults"><span class="std std-ref">default settings</span></a></p>
<hr class="docutils" />
</section>
<<<<<<< HEAD
<section id="assign-taxonomy">
<span id="dada2-taxonomy"></span><h3>ASSIGN TAXONOMY<a class="headerlink" href="#assign-taxonomy" title="Link to this heading"></a></h3>
=======
<section id="assign-taxonomy-asvs-workflow">
<span id="dada2-taxonomy"></span><h4>ASSIGN TAXONOMY [ASVs workflow]<a class="headerlink" href="#assign-taxonomy-asvs-workflow" title="Link to this heading"></a></h4>
>>>>>>> c197e31f5a449b4e3650a9545d50562e2de36a2e
<p>DADA2 <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">assignTaxonomy</a> function to classify ASVs.
Outputs classified fasta files into <code class="docutils literal notranslate"><span class="pre">taxonomy_out.dada2</span></code> directory.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">minBoot</span></code></p></td>
<td><div class="line-block">
<div class="line">the minimum bootstrap confidence for assigning a taxonomic level</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">tryRC</span></code></p></td>
<td><div class="line-block">
<div class="line">the reverse-complement of each sequences will be used for classification</div>
<div class="line">if it is a better match to the reference sequences than the forward sequence</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">dada2</span> <span class="pre">database</span></code></p></td>
<td><div class="line-block">
<div class="line">select a reference database fasta file for taxonomy annotation</div>
<div class="line"><a class="reference external" href="https://benjjneb.github.io/dada2/training.html">Download DADA2-formatted reference databases here</a></div>
</div>
</td>
</tr>
</tbody>
</table>
<p>see <a class="reference internal" href="#dada2-defaults"><span class="std std-ref">default settings</span></a></p>
</section>
</section>
<hr class="docutils" />
<section id="unoise-asvs">
<h2>UNOISE ASVs<a class="headerlink" href="#unoise-asvs" title="Link to this heading"></a></h2>
<<<<<<< HEAD
<p>UNOISE3 pipeline for making ASVs (zOTUs).
Can optionally do automatic clustering of those ASVs (zOTUs) to OTUs by specifying the similarity threshold &lt; 1.
Uses UNOISE3 and clustering algorithms in vsearch.</p>
</section>
<section id="nextits">
<span id="nextits-pipeline"></span><h2>NextITS<a class="headerlink" href="#nextits" title="Link to this heading"></a></h2>
<p><a class="reference external" href="https://next-its.github.io">NextITS</a> is an automated pipeline for analysing <strong>full-length ITS</strong> reads
obtained via <strong>PacBio</strong> sequencing.</p>
<div class="line-block">
<div class="line">This pipeline implements:</div>
<div class="line">* primer trimming</div>
<div class="line">* quality filtering</div>
<div class="line">* full-length ITS region extraction</div>
<div class="line">* correction of homopolymer errors</div>
<div class="line">* chimera filtering (<a class="reference external" href="https://owncloud.ut.ee/owncloud/s/iaQ3i862pjwYgdy">get database for reference-based chimera filtering here</a>)</div>
<div class="line">* recovery of sequences false-positively annotated as chimeric</div>
<div class="line">* detection of tag-switching artifacts per sequencing run</div>
<div class="line">* multiple options for sequence clustering</div>
<div class="line">* post-clustering with LULU</div>
</div>
<p>Please see other details here: <a class="reference external" href="https://next-its.github.io">https://next-its.github.io</a>
<strong>Please note that NextITS pipeline accepts only a single primer pair</strong>, i.e., one forward and one reverse primer in STEP_1!</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>NextITS requires your data and folders to be structured in a specific way (see below)!
Directory <code class="docutils literal notranslate"><span class="pre">my_dir_for_NextITS</span></code> contains <code class="docutils literal notranslate"><span class="pre">Input</span></code> [hard-coded requirement here] and one or multiple sequencing runs.
In the below example, the sequencing runs [<code class="docutils literal notranslate"><span class="pre">RunID</span></code>] are named as Run1, Run2 and Run3 (but naming can be different).</p>
<p>In PipeCraft2, following the examples below, select <code class="docutils literal notranslate"><span class="pre">my_dir_for_NextITS</span></code> as a WORKDIR.</p>
</div>
<section id="single-sequencing-run">
<h3>Single sequencing run<a class="headerlink" href="#single-sequencing-run" title="Link to this heading"></a></h3>
<div class="line-block">
<div class="line">Select <code class="docutils literal notranslate"><span class="pre">my_dir_for_NextITS</span></code> as a WORKDIR in PipeCraft2.</div>
<div class="line">Directory structure for analysing a single sequencing run:</div>
</div>
<p><a class="reference internal" href="_images/NextITS_singledir_structure.png"><img alt="Alternative text" src="_images/NextITS_singledir_structure.png" style="width: 290px;" /></a></p>
<p>Input data for this pipeline <strong>must be demultiplexed</strong>, if your data is multiplexed use the demultiplexer
from <strong>QuickTools</strong> before running the pipeline.</p>
<div class="admonition-sample-naming admonition">
<p class="admonition-title">Sample naming</p>
<p>Please avoid non-ASCII symbols in <code class="docutils literal notranslate"><span class="pre">SampleID</span></code>,
and do not use the period symbol (.), as it represents the wildcard character in regular expressions.
Also, it is preferable not to start the sample name with a number.</p>
</div>
</section>
<section id="multiple-sequencing-runs">
<h3>Multiple sequencing runs<a class="headerlink" href="#multiple-sequencing-runs" title="Link to this heading"></a></h3>
<div class="line-block">
<div class="line">Select <code class="docutils literal notranslate"><span class="pre">my_dir_for_NextITS</span></code> as a WORKDIR in PipeCraft2.</div>
<div class="line">Directory structure for analysing multiple sequencing runs:</div>
</div>
<p><a class="reference internal" href="_images/NextITS_multidir_structure.png"><img alt="Alternative text" src="_images/NextITS_multidir_structure.png" style="width: 350px;" /></a></p>
<p>Input data for this pipeline <strong>must be demultiplexed</strong>, if your data is multiplexed use the demultiplexer
from <strong>QuickTools</strong> before running the pipeline.</p>
<div class="admonition-sample-naming admonition">
<p class="admonition-title">Sample naming</p>
<p>Please avoid non-ASCII symbols in <code class="docutils literal notranslate"><span class="pre">RunID</span></code> and <code class="docutils literal notranslate"><span class="pre">SampleID</span></code>,
and do not use the period symbol (.), as it represents the wildcard character in regular expressions.
Also, it is preferable not to start the sample name with a number.</p>
</div>
<p>NextITS uses the <code class="docutils literal notranslate"><span class="pre">SequencingRunID__SampleID</span></code> naming convention (please note the double underscore separating <code class="docutils literal notranslate"><span class="pre">RunID</span></code> and <code class="docutils literal notranslate"><span class="pre">SampleID</span></code> parts).
This naming scheme allows to easily trace back sequences, especially if the same sample was sequenced several times and is present in multiple sequencing runs.
In the later steps, extracting the SampleID part and summarizing read counts for such samples is easy.</p>
<p><strong>Default settings:</strong></p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Analyses step</p></th>
<th class="head"><p>Default setting</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><div class="line-block">
<div class="line">STEP 1: <a class="reference external" href="https://next-its.github.io/assets/NextITS_Workflow_Step1.webp">QUALITY CONTROL, ARTEFACT REMOVAL</a></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
</td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">primer_mismatch</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">its_region</span></code> = full</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">qc_maxhomopolymerlen</span></code> = 25</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">qc_maxn</span></code> = 4</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">ITSx_evalue</span></code> = 1e-2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">ITSx_partial</span></code> = 0</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">ITSx_tax</span></code> = all</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">chimera_rescue_occurrence</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">tj</span> <span class="pre">f</span></code> = 0.01</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">tj</span> <span class="pre">p</span></code> = 1</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">hp</span></code> = TRUE</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><div class="line-block">
<div class="line">STEP 2: <a class="reference external" href="https://next-its.github.io/assets/NextITS_Workflow_Step2.webp">DATA AGGREGATION, CLUSTERING</a></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
</td>
<td><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">otu_id</span></code> = 0.98</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">swarm_d</span></code> = 1</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">lulu</span></code> = TRUE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">unoise</span></code> = FALSE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">otu_id_def</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">otu_qmask</span></code> = dust</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">swarm_fastidious</span></code> = TRUE</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">unoise_alpha</span></code> = 2</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">unoise_minsize</span></code> = 8</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">max_MEEP</span></code> = 0.5</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">max_chimera_score</span></code> = 0.5</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">lulu_match</span></code> = 95</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">lulu_ratio</span></code> = 1</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">lulu_ratiotype</span></code> = min</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">lulu_relcooc</span></code> = 0.95</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">lulu_maxhits</span></code> = 0</div>
</div>
</td>
</tr>
</tbody>
</table>
</section>
<section id="cut-primers">
<h3>Cut primers<a class="headerlink" href="#cut-primers" title="Link to this heading"></a></h3>
<p><strong>Please note that NextITS pipeline accepts only a single primer pair</strong>, i.e., one forward and one reverse primer!</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">primer_forward</span></code></p></td>
<td><div class="line-block">
<div class="line">Specify forward primer, IUPAC codes allowed</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">primer_reverse</span></code></p></td>
<td><div class="line-block">
<div class="line">Specify reverse primer, IUPAC codes allowed</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">primer_mismatch</span></code></p></td>
<td><div class="line-block">
<div class="line">Specify allowed number of mismatches for primers</div>
</div>
</td>
</tr>
</tbody>
</table>
</section>
<section id="id2">
<h3>Quality filtering<a class="headerlink" href="#id2" title="Link to this heading"></a></h3>
<p>Filter sequences based on expected errors per sequence and per base, compress and correct homopolymers.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">qc_maxee</span></code></p></td>
<td><div class="line-block">
<div class="line">Maximum number of expected errors</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">qc_maxeerate</span></code></p></td>
<td><div class="line-block">
<div class="line">Maximum number of expected error per base</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">qc_maxn</span></code></p></td>
<td><div class="line-block">
<div class="line">Discard sequences with more than the specified number of ambiguous nucleotides (N’s)</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">qc_maxhomopolymerlen</span></code></p></td>
<td><div class="line-block">
<div class="line">Threshold for a homopolymer region lenght in a sequence</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">hp</span></code></p></td>
<td><div class="line-block">
<div class="line">Enable or disable homopolymer correction</div>
</div>
</td>
</tr>
</tbody>
</table>
</section>
<section id="its-extraction">
<h3>ITS extraction<a class="headerlink" href="#its-extraction" title="Link to this heading"></a></h3>
<div class="line-block">
<div class="line">When performing ITS metabarcoding, it may be beneficial to trim the flanking 18S and 28S rRNA genes; because:</div>
</div>
<blockquote>
<div><ul class="simple">
<li><p>these conserved regions don’t offer species-level differentiation.</p></li>
<li><p>random errors in these areas can disrupt sequence clustering.</p></li>
<li><p>chimeric breakpoints, which are common in these regions, are hard to detect in short fragments ranging from 10 to 70 bases.</p></li>
</ul>
<div class="line-block">
<div class="line">NextITS deploys the <a class="reference external" href="https://microbiology.se/software/itsx/">ITSx software</a> (<a class="reference external" href="https://doi.org/10.1111/2041-210X.12073">Bengtsson-Palme et al. 2013</a>) for extracting the ITS sequence.</div>
</div>
</div></blockquote>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">its_region</span></code></p></td>
<td><div class="line-block">
<div class="line">ITS part selector (ITS1, ITS2 or full)</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">ITSx_tax</span></code></p></td>
<td><div class="line-block">
<div class="line">Taxonomy profile for ITSx can be used to restrict the search to only taxon(s) of interest.</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">ITSx_evalue</span></code></p></td>
<td><div class="line-block">
<div class="line">E-value cutoff threshold for ITSx</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">ITSx_partial</span></code></p></td>
<td><div class="line-block">
<div class="line">Keep partial ITS sequences (specify a minimum length cutoff)</div>
</div>
</td>
</tr>
</tbody>
</table>
</section>
<section id="id3">
<h3>Chimera filtering<a class="headerlink" href="#id3" title="Link to this heading"></a></h3>
<div class="line-block">
<div class="line">NextITS employs a two-pronged strategy to detect chimeras: de novo and reference-based chimera filtering.</div>
<div class="line">A <strong>reference database</strong> for chimera filtering from full-length ITS data is <a class="reference external" href="https://owncloud.ut.ee/owncloud/s/iaQ3i862pjwYgdy">accessible here</a>. This database is based on <a class="reference external" href="https://eukaryome.org">EUKARYOME database</a></div>
</div>
<p>Additional step in NextITS is a <strong>“rescue” of sequences</strong> that were initially flagged as chimeric, but are occur at least in 2 samples (which represent independent PCR reactions);
thus are likely false-positive chimeric sequences. The chimeric sequence occurrence frequency can be edited using the –chimera_rescue_occurrence parameter.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">chimera_database</span></code> (optional)</p></td>
<td><div class="line-block">
<div class="line">Database for reference based chimera removal (UDB)</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">chimera_rescue_occurence</span></code></p></td>
<td><div class="line-block">
<div class="line">A minimum occurence of initially flagged chimeric sequence required to rescue them</div>
</div>
</td>
</tr>
</tbody>
</table>
</section>
<section id="tag-jump-correction">
<h3>Tag-jump correction<a class="headerlink" href="#tag-jump-correction" title="Link to this heading"></a></h3>
<blockquote>
<div><p>Tag-jumps, sometimes referred to as index-switches or index cross-talk, may represent a significant concern in high-throughput sequencing (HTS) data.
They can cause technical cross-contamination between samples, potentially distorting estimates of community composition.
Here, tag-jump events are evaluated the UNCROSS2 algorithm (<a class="reference external" href="https://www.biorxiv.org/content/10.1101/400762v1">Edgar 2018</a> ) are removed.</p>
</div></blockquote>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">tj_f</span></code></p></td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://www.drive5.com/usearch/manual/uncross_algo.html">UNCROSS</a> parameter f for tag-jump filtering</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">tj_p</span></code></p></td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://www.drive5.com/usearch/manual/uncross_algo.html">UNCROSS</a> parameter p for tag-jump filtering</div>
</div>
</td>
</tr>
</tbody>
</table>
</section>
<section id="unoise-denoising">
<h3>UNOISE denoising<a class="headerlink" href="#unoise-denoising" title="Link to this heading"></a></h3>
<blockquote>
<div><div class="line-block">
<div class="line">The UNOISE algorithm (<a class="reference external" href="https://www.biorxiv.org/content/10.1101/081257v1">Edgar 2016</a> ) focuses on error-correction (or denoising) of amplicon reads. Essentially, UNOISE operates on the principle that if a sequence with low abundance closely resembles another sequence with high abundance, the former is probably an error. This helps differentiate between true biological variation and sequencing errors. It’s important to note that UNOISE was initially designed and optimized for Illumina data. Because of indel errors stemming from inaccuracies in homopolymeric regions, UNOISE might not work well with data that hasn’t undergone homopolymer correction.</div>
</div>
</div></blockquote>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">unoise</span></code></p></td>
<td><div class="line-block">
<div class="line">Enable or disable denoising with <a class="reference external" href="https://www.drive5.com/usearch/manual/unoise_algo.html">UNOISE</a> algorithm</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">unoise_alpha</span></code></p></td>
<td><div class="line-block">
<div class="line">Alpha parameter for <a class="reference external" href="https://www.drive5.com/usearch/manual/unoise_algo.html">UNOISE</a></div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">unoise_minsize</span></code></p></td>
<td><div class="line-block">
<div class="line">Minimum sequence abundance</div>
</div>
</td>
</tr>
</tbody>
</table>
</section>
<section id="clustering">
<h3>Clustering<a class="headerlink" href="#clustering" title="Link to this heading"></a></h3>
<p>NextITS supports 3 different clustering methods:</p>
<blockquote>
<div><ul class="simple">
<li><p>vsearch:
this employs greedy clustering using a fixed sequence similarity threshold with VSEARCH (<a class="reference external" href="https://peerj.com/articles/2584/">Rognes et al., 2016,</a> );</p></li>
<li><p>swarm:
dynamic sequence similarity threshold for clustering with SWARM (<a class="reference external" href="https://academic.oup.com/bioinformatics/article/38/1/267/6318385?login=false">Mahé et al., 2021,</a> );</p></li>
<li><p>unoise:
creates zero-radius OTUs (zOTUs) based on the UNOISE3 algorithm (<a class="reference external" href="https://www.biorxiv.org/content/10.1101/081257v1">Edgar 2016</a> );</p></li>
</ul>
</div></blockquote>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">clustering_method</span></code></p></td>
<td><div class="line-block">
<div class="line">Sequence clustering method (choose from: vsearch, swarm, unoise)</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">otu_id</span></code></p></td>
<td><div class="line-block">
<div class="line">Sequence similarity threshold</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">otu_iddef</span></code></p></td>
<td><div class="line-block">
<div class="line">Sequence similarity definition (applied to UNOISE as well)</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">otu_qmask</span></code></p></td>
<td><div class="line-block">
<div class="line">Method to mask low-complexity sequences (applied to UNOISE as well)</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">swarm_d</span></code></p></td>
<td><div class="line-block">
<div class="line"><a class="reference external" href="https://github.com/torognes/swarm">SWARM</a> clustering resolution (d)</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">swarm_fastidious</span></code></p></td>
<td><div class="line-block">
<div class="line">Link nearby low-abundance swarms (fastidious option)</div>
</div>
</td>
</tr>
</tbody>
</table>
</section>
<section id="post-clustering-with-lulu">
<h3>Post-clustering with LULU<a class="headerlink" href="#post-clustering-with-lulu" title="Link to this heading"></a></h3>
<p>The purpose of LULU is to reduce the number of erroneous OTUs in OTU tables to achieve more realistic biodiversity metrics.
By evaluating the co-occurence patterns of OTUs among samples LULU identifies OTUs that
consistently satisfy some user selected criteria for being errors of more abundant OTUs and merges these OTUs.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Setting</p></th>
<th class="head"><p>Tooltip</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">lulu</span></code></p></td>
<td><div class="line-block">
<div class="line">Enable or disable post-clustering curation with <a class="reference external" href="https://github.com/tobiasgf/lulu">lulu</a></div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">lulu_match</span></code></p></td>
<td><div class="line-block">
<div class="line">Minimum similarity threshold</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">lulu_ratio</span></code></p></td>
<td><div class="line-block">
<div class="line">Minimum abundance ratio</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">lulu_ratiotype</span></code></p></td>
<td><div class="line-block">
<div class="line">Abundance ratio type - “min” or “avg</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">lulu_relcooc</span></code></p></td>
<td><div class="line-block">
<div class="line">Relative co-occurrence</div>
</div>
</td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">lulu_maxhits</span></code></p></td>
<td><div class="line-block">
<div class="line">Maximum number of hits (0 = unlimited)</div>
</div>
</td>
</tr>
</tbody>
</table>
</section>
=======
<p>UNOISE3 pipeline for making ASVs (zOTUs) + optionally automatic clustering of those ASVs.
Updating this section soon.</p>
</section>
<section id="nextits">
<span id="id1"></span><h2>NextITS<a class="headerlink" href="#nextits" title="Link to this heading"></a></h2>
<p>NextITS pipeline for PacBio ITS sequences.
Updating this section soon.</p>
>>>>>>> c197e31f5a449b4e3650a9545d50562e2de36a2e
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="quickstart.html" class="btn btn-neutral float-left" title="QuickStart" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="quicktools.html" class="btn btn-neutral float-right" title="Individual steps (Quick Tools)" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Sten Anslan.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>