<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Releases &mdash; PipeCraft2 1.0.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=8d563738"></script>
        <script src="_static/doctools.js?v=9a2dae69"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Docker images" href="docker_images.html" />
    <link rel="prev" title="How to cite" href="citation.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            PipeCraft2
          </a>
              <div class="version">
                1.0.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="quickstart.html">QuickStart</a></li>
<li class="toctree-l1"><a class="reference internal" href="pre-defined_pipelines.html">Pre-defined pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="quicktools.html">Individual steps (Quick Tools)</a></li>
<li class="toctree-l1"><a class="reference internal" href="postprocessing.html">Post-processing tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorial.html">Example data analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="troubleshoot.html">Troubleshooting</a></li>
<li class="toctree-l1"><a class="reference internal" href="licence.html">Licence</a></li>
<li class="toctree-l1"><a class="reference internal" href="contact.html">Contact and Acknowledgements</a></li>
<li class="toctree-l1"><a class="reference internal" href="citation.html">How to cite</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Releases</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#id2">1.0.0 (01.09.2023)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#id5">0.1.4 (15.12.2022)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#id21">0.1.3 (28.07.2022)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#id23">0.1.2 (07.06.2022)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#id53">0.1.1 (01.04.2022)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#pre-release-14-12-2021">0.1.0 pre-release (14.12.2021)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="docker_images.html">Docker images</a></li>
<li class="toctree-l1"><a class="reference internal" href="for_developers.html">For Developers</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">PipeCraft2</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Releases</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/releases.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <dl class="simple">
<dt><a class="reference internal" href="_images/PipeCraft2_icon_v2.png"><img alt="Alternative text" src="_images/PipeCraft2_icon_v2.png" style="width: 100px;" /></a></dt><dd><p><a class="reference external" href="https://github.com/pipecraft2/pipecraft">github</a></p>
</dd>
</dl>
<style> .red {color:#ff0000; font-weight:bold; font-size:16px} </style><section id="releases">
<span id="id1"></span><h1><a class="toc-backref" href="#id78" role="doc-backlink">Releases</a><a class="headerlink" href="#releases" title="Link to this heading"></a></h1>
<nav class="contents" id="contents">
<p class="topic-title">Contents</p>
<ul class="simple">
<li><p><a class="reference internal" href="#releases" id="id78">Releases</a></p>
<ul>
<li><p><a class="reference internal" href="#id2" id="id79">1.0.0 (01.09.2023)</a></p></li>
<li><p><a class="reference internal" href="#id5" id="id80">0.1.4 (15.12.2022)</a></p></li>
<li><p><a class="reference internal" href="#id21" id="id81">0.1.3 (28.07.2022)</a></p></li>
<li><p><a class="reference internal" href="#id23" id="id82">0.1.2 (07.06.2022)</a></p></li>
<li><p><a class="reference internal" href="#id53" id="id83">0.1.1 (01.04.2022)</a></p></li>
<li><p><a class="reference internal" href="#pre-release-14-12-2021" id="id84">0.1.0 pre-release (14.12.2021)</a></p></li>
</ul>
</li>
</ul>
</nav>
<hr class="docutils" />
<section id="id2">
<span id="id3"></span><h2><a class="toc-backref" href="#id79" role="doc-backlink">1.0.0 (01.09.2023)</a><a class="headerlink" href="#id2" title="Link to this heading"></a></h2>
<p><a class="reference external" href="https://github.com/pipecraft2/pipecraft/releases/tag/v1.0.0">DOWNLOAD link for v1.0.0</a></p>
<ul class="simple">
<li><p>major updates in the front-end; individual tools on the right, pipelines on left.</p></li>
<li><p>added debugging mode and improved log info</p></li>
<li><p>added NextITS pipeline for PacBio ITS sequences (not available for MacOS release)</p></li>
<li><p>added ORFfinder + HMM bsed pseudogene/off-targets filtering for protein coding genes</p></li>
<li><p>added RDP classifier</p></li>
<li><p>added DADA2 pipeline for PacBio data</p></li>
<li><p>added DADA2 pipeline for paired-end mixed oriented amplicons (fwd_orient and rev_orient are denoised separately and then merged)</p></li>
<li><p>implemented DADA2 denoising sensitivity editing</p></li>
<li><p>all features will get sha1 ID</p></li>
<li><p>added ASVs to OTUs module (cluster ASVs into OTUs with vsearch)</p></li>
<li><p>added tag-jumps filtering module (UNCROSS2)</p></li>
<li><p>fixed the vsearch_dada2 container issues for MacOS</p></li>
</ul>
<p>Implemented software:
<em>(software in red font denote new additions; ‘version’ in bold denotes version upgrade)</em></p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Software</p></th>
<th class="head"><p>version</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><span class="red">NextITS pipeline</span> <a class="reference external" href="https://next-its.github.io/">(link)</a></p></td>
<td><p><strong>0.5.0</strong></p></td>
</tr>
<tr class="row-odd"><td><p><span class="red">ORFfinder</span> <a class="reference external" href="https://www.ncbi.nlm.nih.gov/orffinder/">(link)</a></p></td>
<td><p><strong>v0.4.3</strong></p></td>
</tr>
<tr class="row-even"><td><p><span class="red">RDP classifier</span></p></td>
<td><p><strong>v2.13</strong></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://benjjneb.github.io/dada2/index.html">DADA2</a></p></td>
<td><p><strong>1.27</strong></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://github.com/torognes/vsearch">vsearch</a></p></td>
<td><p><strong>2.23</strong></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="http://www.usadellab.org/cms/?page=trimmomatic">trimmomatic</a></p></td>
<td><p>0.39</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://bioinf.shenwei.me/seqkit/">seqkit</a></p></td>
<td><p>2.3.0</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://cutadapt.readthedocs.io/en/stable/">cutadapt</a></p></td>
<td><p><strong>4.4</strong></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://github.com/mothur/mothur">mothur</a></p></td>
<td><p>1.46.1</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://microbiology.se/software/itsx/">ITS Extractor</a></p></td>
<td><p>1.1.3</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://github.com/indraniel/fqgrep">fqgrep</a></p></td>
<td><p>0.4.4</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a></p></td>
<td><p><strong>2.14.0+</strong></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></p></td>
<td><p>0.11.9</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://multiqc.info/">MultiQC</a></p></td>
<td><p>1.12</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://github.com/tobiasgf/lulu">LULU</a></p></td>
<td><p>0.1.0</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/OpenGene/fastp">fastp</a></p></td>
<td><p>0.23.2</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://github.com/biocore/DEICODE">DEICODE</a></p></td>
<td><p>0.2.4</p></td>
</tr>
</tbody>
</table>
<hr class="docutils" />
</section>
<section id="id5">
<span id="id6"></span><h2><a class="toc-backref" href="#id80" role="doc-backlink">0.1.4 (15.12.2022)</a><a class="headerlink" href="#id5" title="Link to this heading"></a></h2>
<p><a class="reference external" href="https://github.com/pipecraft2/pipecraft/releases/tag/v0.1.4">DOWNLOAD link for v0.1.4</a></p>
<ul class="simple">
<li><p>added 2nd round of cut primers to properly remove fwd and rev primers form the paired-end data set</p></li>
<li><p>added UNOISE3 module to generate zOTUs (under clustering)</p></li>
<li><p>added uchime3 chimera filtering (for denoised amplicons)</p></li>
<li><p>edited sequence count statistics process after the process (using seqkit)</p></li>
<li><p>only fasta (fa, fas) format is accepted for clustering</p></li>
<li><p>edited OTU table making strategy for OTU clustering (was –usearch_global before)</p></li>
<li><p>added table filtering options for DADA2 ASV table (collapse mismatch, filter by length)</p></li>
<li><p>added ASV to OTU module (clustering DADA2 ASVs into OTUs)</p></li>
<li><p>select region to cluster after ITSx in OTUs workflow</p></li>
<li><p>automatically saves the PipeCraft workflow settings into loadable JSON file</p></li>
<li><p>outputs log file (in development)</p></li>
<li><p>merged vsearch and dada2 containers (had a lot in common)</p></li>
</ul>
<p>Implemented software:
<em>(software version in bold denotes version upgrade)</em></p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Software</p></th>
<th class="head"><p>version</p></th>
<th class="head"><p>Reference</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://benjjneb.github.io/dada2/index.html">DADA2</a></p></td>
<td><p>1.20</p></td>
<td><p><a class="reference external" href="https://www.nature.com/articles/nmeth.3869">Callahan et. al 2016</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/torognes/vsearch">vsearch</a></p></td>
<td><p><strong>2.22.1</strong></p></td>
<td><p><a class="reference external" href="https://peerj.com/articles/2584/">Rognes et. al 2016</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="http://www.usadellab.org/cms/?page=trimmomatic">trimmomatic</a></p></td>
<td><p>0.39</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btu170">Bolger et al. 2014</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://bioinf.shenwei.me/seqkit/">seqkit</a></p></td>
<td><p><strong>2.3.0</strong></p></td>
<td><p><a class="reference external" href="https://doi.org/10.1371/journal.pone.0163962">Shen et al. 2016</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://cutadapt.readthedocs.io/en/stable/">cutadapt</a></p></td>
<td><p>3.5</p></td>
<td><p><a class="reference external" href="https://doi.org/10.14806/ej.17.1.200">Martin 2011</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/mothur/mothur">mothur</a></p></td>
<td><p>1.46.1</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1128/AEM.01541-09">Schloss et al. 2009</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://microbiology.se/software/itsx/">ITS Extractor</a></p></td>
<td><p>1.1.3</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1111/2041-210X.12073">Bengtsson-Palme et al. 2013</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/indraniel/fqgrep">fqgrep</a></p></td>
<td><p>0.4.4</p></td>
<td><p><a class="reference external" href="https://github.com/indraniel/fqgrep">Indraniel Das 2011</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a></p></td>
<td><p>2.11.0+</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1186/1471-2105-10-421">Camacho et al. 2009</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></p></td>
<td><p>0.11.9</p></td>
<td><p><a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">Andrews 2019</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://multiqc.info/">MultiQC</a></p></td>
<td><p>1.12</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btw354">Ewels et al. 2016</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/tobiasgf/lulu">LULU</a></p></td>
<td><p>0.1.0</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1038/s41467-017-01312-x">Froslev et al. 2017</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://github.com/OpenGene/fastp">fastp</a></p></td>
<td><p>0.23.2</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/bty560">Chen et al. 2018</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/biocore/DEICODE">DEICODE</a></p></td>
<td><p>0.2.4</p></td>
<td><p><a class="reference external" href="https://journals.asm.org/doi/10.1128/mSystems.00016-19">Martion et al. 2019</a></p></td>
</tr>
</tbody>
</table>
<hr class="docutils" />
</section>
<section id="id21">
<span id="id22"></span><h2><a class="toc-backref" href="#id81" role="doc-backlink">0.1.3 (28.07.2022)</a><a class="headerlink" href="#id21" title="Link to this heading"></a></h2>
<p><a class="reference external" href="https://github.com/SuvalineVana/pipecraft/releases/tag/v0.1.3">DOWNLOAD link for v0.1.3</a></p>
<ul class="simple">
<li><p>updated BLAST 2.11.0+ to BLAST 2.12.0+ and added biopython to BLAST container (fixed the coverage% calculation)</p></li>
<li><p>fixed the megaBLAST, when gapextend=undefined</p></li>
<li><p>quality Check module edit (does not stop when browsing around)</p></li>
<li><p>fixed ASVs workflow error message when using &lt;2 samples</p></li>
<li><p>added lock panels when starting a process</p></li>
<li><p>few cosmetic front-end adds</p></li>
</ul>
</section>
<section id="id23">
<span id="id24"></span><h2><a class="toc-backref" href="#id82" role="doc-backlink">0.1.2 (07.06.2022)</a><a class="headerlink" href="#id23" title="Link to this heading"></a></h2>
<p><a class="reference external" href="https://github.com/SuvalineVana/pipecraft/releases/tag/v0.1.2">DOWNLOAD link for v0.1.2</a></p>
<ul class="simple">
<li><p>added LULU post-clustering</p></li>
<li><p>added DEICODE (postprocessing)</p></li>
<li><p>added fastp quality filtering</p></li>
<li><p>added DADA2 quality filtering under ‘ADD STEP’ -&gt; ‘QUALITY FILTERING’ panel</p></li>
<li><p>added DADA2 denoise and assemble paired-end data under ‘ADD STEP’ -&gt; ‘ASSEMBLE PAIRED-END’ panel</p></li>
<li><p>added DADA2 assignTaxonomy under ‘ADD STEP’ -&gt; ‘ASSIGN TAXONOMY’ panel</p></li>
<li><p>added trunc_length option for vsearch quality filtering</p></li>
<li><p>python3 module fix for ITSx for removing empty sequeces</p></li>
</ul>
<p>Implemented software:
<em>(software in red font denote new additions; ‘version’ in bold denotes version upgrade)</em></p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Software</p></th>
<th class="head"><p>version</p></th>
<th class="head"><p>Reference</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://benjjneb.github.io/dada2/index.html">DADA2</a></p></td>
<td><p><strong>1.20</strong></p></td>
<td><p><a class="reference external" href="https://www.nature.com/articles/nmeth.3869">Callahan et. al 2016</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/torognes/vsearch">vsearch</a></p></td>
<td><p>2.18.0</p></td>
<td><p><a class="reference external" href="https://peerj.com/articles/2584/">Rognes et. al 2016</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="http://www.usadellab.org/cms/?page=trimmomatic">trimmomatic</a></p></td>
<td><p>0.39</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btu170">Bolger et al. 2014</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://bioinf.shenwei.me/seqkit/">seqkit</a></p></td>
<td><p>2.0.0</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1371/journal.pone.0163962">Shen et al. 2016</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://cutadapt.readthedocs.io/en/stable/">cutadapt</a></p></td>
<td><p>3.5</p></td>
<td><p><a class="reference external" href="https://doi.org/10.14806/ej.17.1.200">Martin 2011</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/mothur/mothur">mothur</a></p></td>
<td><p>1.46.1</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1128/AEM.01541-09">Schloss et al. 2009</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://microbiology.se/software/itsx/">ITS Extractor</a></p></td>
<td><p>1.1.3</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1111/2041-210X.12073">Bengtsson-Palme et al. 2013</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/indraniel/fqgrep">fqgrep</a></p></td>
<td><p>0.4.4</p></td>
<td><p><a class="reference external" href="https://github.com/indraniel/fqgrep">Indraniel Das 2011</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a></p></td>
<td><p>2.11.0+</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1186/1471-2105-10-421">Camacho et al. 2009</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></p></td>
<td><p>0.11.9</p></td>
<td><p><a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">Andrews 2019</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://multiqc.info/">MultiQC</a></p></td>
<td><p>1.12</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btw354">Ewels et al. 2016</a></p></td>
</tr>
<tr class="row-odd"><td><p><span class="red">LULU</span> <a class="reference external" href="https://github.com/tobiasgf/lulu">(link)</a></p></td>
<td><p>0.1.0</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1038/s41467-017-01312-x">Froslev et al. 2017</a></p></td>
</tr>
<tr class="row-even"><td><p><span class="red">fastp</span> <a class="reference external" href="https://github.com/OpenGene/fastp">(link)</a></p></td>
<td><p>0.23.2</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/bty560">Chen et al. 2018</a></p></td>
</tr>
<tr class="row-odd"><td><p><span class="red">DEICODE</span> <a class="reference external" href="https://github.com/biocore/DEICODE">(link)</a></p></td>
<td><p>0.2.4</p></td>
<td><p><a class="reference external" href="https://journals.asm.org/doi/10.1128/mSystems.00016-19">Martion et al. 2019</a></p></td>
</tr>
</tbody>
</table>
<hr class="docutils" />
</section>
<section id="id53">
<span id="id54"></span><h2><a class="toc-backref" href="#id83" role="doc-backlink">0.1.1 (01.04.2022)</a><a class="headerlink" href="#id53" title="Link to this heading"></a></h2>
<p>Minor cosmetic changes and bug fixes.
<a class="reference external" href="https://github.com/SuvalineVana/pipecraft/releases/tag/0.1.1">DOWNLOAD link for v0.1.1</a></p>
<ul class="simple">
<li><p>separate output forlder for unused index combinations in demultiplexing.</p></li>
<li><p>resolved issues with sample renaiming when using dual combinational indexes for paired-end data
(DEMULTIPLEX)</p></li>
<li><p>minBoot option fixed in DADA2 taxonomy annotation</p></li>
<li><p>vsearch quality filtering “minsize” not working (option currently removed).</p></li>
</ul>
<hr class="docutils" />
</section>
<section id="pre-release-14-12-2021">
<span id="id55"></span><h2><a class="toc-backref" href="#id84" role="doc-backlink">0.1.0 pre-release (14.12.2021)</a><a class="headerlink" href="#pre-release-14-12-2021" title="Link to this heading"></a></h2>
<p><a class="reference external" href="https://github.com/SuvalineVana/pipecraft/releases/tag/0.1.0">DOWNLOAD link for v0.1.0</a></p>
<ul class="simple">
<li><p>ASV workflow with DADA2 for paired-end data.</p></li>
<li><p>vsearch based OTU workflow.</p></li>
<li><p>QualityCheck module with MultiQC and FastQC</p></li>
</ul>
<p>Implemented software:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Software</p></th>
<th class="head"><p>version</p></th>
<th class="head"><p>Reference</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://benjjneb.github.io/dada2/index.html">DADA2</a></p></td>
<td><p>1.14</p></td>
<td><p><a class="reference external" href="https://www.nature.com/articles/nmeth.3869">Callahan et. al 2016</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/torognes/vsearch">vsearch</a></p></td>
<td><p>2.18.0</p></td>
<td><p><a class="reference external" href="https://peerj.com/articles/2584/">Rognes et. al 2016</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="http://www.usadellab.org/cms/?page=trimmomatic">trimmomatic</a></p></td>
<td><p>0.39</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btu170">Bolger et al. 2014</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://bioinf.shenwei.me/seqkit/">seqkit</a></p></td>
<td><p>2.0.0</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1371/journal.pone.0163962">Shen et al. 2016</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://cutadapt.readthedocs.io/en/stable/">cutadapt</a></p></td>
<td><p>3.5</p></td>
<td><p><a class="reference external" href="https://doi.org/10.14806/ej.17.1.200">Martin 2011</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/mothur/mothur">mothur</a></p></td>
<td><p>1.46.1</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1128/AEM.01541-09">Schloss et al. 2009</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://microbiology.se/software/itsx/">ITS Extractor</a></p></td>
<td><p>1.1.3</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1111/2041-210X.12073">Bengtsson-Palme et al. 2013</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/indraniel/fqgrep">fqgrep</a></p></td>
<td><p>0.4.4</p></td>
<td><p><a class="reference external" href="https://github.com/indraniel/fqgrep">Indraniel Das 2011</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a></p></td>
<td><p>2.11.0+</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1186/1471-2105-10-421">Camacho et al. 2009</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></p></td>
<td><p>0.11.9</p></td>
<td><p><a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">Andrews 2019</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://multiqc.info/">MultiQC</a></p></td>
<td><p>1.12</p></td>
<td><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btw354">Ewels et al. 2016</a></p></td>
</tr>
</tbody>
</table>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="citation.html" class="btn btn-neutral float-left" title="How to cite" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="docker_images.html" class="btn btn-neutral float-right" title="Docker images" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Sten Anslan.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>