<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />
<meta content="PipeCraft manual. tutorial" lang="en" name="description" xml:lang="en" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Example data analyses &mdash; PipeCraft2 1.0.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Troubleshooting" href="troubleshoot.html" />
    <link rel="prev" title="Post-processing tools" href="postprocessing.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> PipeCraft2
          </a>
              <div class="version">
                1.0.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="overview.html">General overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="pre-defined_pipelines.html">Pre-defined pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="quicktools.html">Individual steps (Quick Tools)</a></li>
<li class="toctree-l1"><a class="reference internal" href="postprocessing.html">Post-processing tools</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Example data analyses</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#inspect-quality-profiles">Inspect quality profiles</a></li>
<li class="toctree-l2"><a class="reference internal" href="#paired-end-illumina-or-mgi-tech-data">Paired-end Illumina (or MGI-Tech) data</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#demultiplexed-paired-end-data-asv-workflow-with-dada2">Demultiplexed paired-end data; ASV workflow with DADA2</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#examine-the-outputs">Examine the outputs</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#demultiplexed-paired-end-data-otu-workflow">Demultiplexed paired-end data; OTU workflow</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#id3">Examine the outputs</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#multiplexed-library">Multiplexed library</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#single-end-pacbio-or-assembled-paired-end-data">Single-end (PacBio or assembled paired-end) data</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="troubleshoot.html">Troubleshooting</a></li>
<li class="toctree-l1"><a class="reference internal" href="licence.html">Licence</a></li>
<li class="toctree-l1"><a class="reference internal" href="contact.html">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="citation.html">How to cite</a></li>
<li class="toctree-l1"><a class="reference internal" href="releases.html">Releases</a></li>
<li class="toctree-l1"><a class="reference internal" href="docker_images.html">Docker images</a></li>
<li class="toctree-l1"><a class="reference internal" href="for_developers.html">For Developers</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">PipeCraft2</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Example data analyses</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/tutorial.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <style> .red {color:#ff0000; font-weight:bold; font-size:16px} </style><style> .green {color:#00f03d; font-weight:bold; font-size:16px} </style><dl class="simple">
<dt><a class="reference internal" href="_images/PipeCraft2_icon_v2.png"><img alt="Alternative text" src="_images/PipeCraft2_icon_v2.png" style="width: 100px;" /></a></dt><dd><p><a class="reference external" href="https://github.com/pipecraft2/pipecraft">github</a></p>
</dd>
</dl>
<section id="example-data-analyses">
<h1><a class="toc-backref" href="#id4">Example data analyses</a><a class="headerlink" href="#example-data-analyses" title="Permalink to this headline"></a></h1>
<p>Some example analyses pipelines.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>When samples of interest are distributed between different sequencing libraries, then first demultiplex (if needed)
libraries separately and place samples of interest into separate working directory.</p>
</div>
<div class="contents topic" id="contents">
<p class="topic-title">Contents</p>
<ul class="simple">
<li><p><a class="reference internal" href="#example-data-analyses" id="id4">Example data analyses</a></p>
<ul>
<li><p><a class="reference internal" href="#inspect-quality-profiles" id="id5">Inspect quality profiles</a></p></li>
<li><p><a class="reference internal" href="#paired-end-illumina-or-mgi-tech-data" id="id6">Paired-end Illumina (or MGI-Tech) data</a></p>
<ul>
<li><p><a class="reference internal" href="#demultiplexed-paired-end-data-asv-workflow-with-dada2" id="id7">Demultiplexed paired-end data; ASV workflow with DADA2</a></p>
<ul>
<li><p><a class="reference internal" href="#examine-the-outputs" id="id8">Examine the outputs</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#demultiplexed-paired-end-data-otu-workflow" id="id9">Demultiplexed paired-end data; OTU workflow</a></p>
<ul>
<li><p><a class="reference internal" href="#id3" id="id10">Examine the outputs</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#multiplexed-library" id="id11">Multiplexed library</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#single-end-pacbio-or-assembled-paired-end-data" id="id12">Single-end (PacBio or assembled paired-end) data</a></p></li>
</ul>
</li>
</ul>
</div>
<hr class="docutils" />
<section id="inspect-quality-profiles">
<h2><a class="toc-backref" href="#id5">Inspect quality profiles</a><a class="headerlink" href="#inspect-quality-profiles" title="Permalink to this headline"></a></h2>
<p>Examine the quality profiles and basic statistics of the your data set using <strong>QualityCheck module</strong>. <a class="reference internal" href="overview.html#qualitycheck"><span class="std std-ref">See here</span></a>.</p>
<hr class="docutils" />
<div class="line-block">
<div class="line"><br /></div>
</div>
</section>
<section id="paired-end-illumina-or-mgi-tech-data">
<h2><a class="toc-backref" href="#id6">Paired-end Illumina (or MGI-Tech) data</a><a class="headerlink" href="#paired-end-illumina-or-mgi-tech-data" title="Permalink to this headline"></a></h2>
<p>Example analyses of paired-end data. Starting with raw paired-end fastq files, finishing with ASV/OTU table and taxonomy table.</p>
<section id="demultiplexed-paired-end-data-asv-workflow-with-dada2">
<span id="demultiplexed-pe-asv"></span><h3><a class="toc-backref" href="#id7">Demultiplexed paired-end data; ASV workflow with DADA2</a><a class="headerlink" href="#demultiplexed-paired-end-data-asv-workflow-with-dada2" title="Permalink to this headline"></a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This tutorial follows <a class="reference external" href="https://benjjneb.github.io/dada2/tutorial.html">DADA2 Pipeline Tutorial</a>.</p>
</div>
<p>Here, we perform example analyses of paired-end data using <a class="reference external" href="https://mothur.org/wiki/miseq_sop/">mothur MiSeq SOP example data set</a>.
<a class="reference external" href="https://mothur.s3.us-east-2.amazonaws.com/wiki/miseqsopdata.zip">Download example data set here</a> (35.1 Mb) and unzip it.
This data set represents demultiplexed set (per-sample fastq files) of 16S rRNA gene V4 amplicon sequences where sample indexes and primers have already been removed.</p>
<ul class="simple">
<li><p>If working with multiplexed data, <a class="reference internal" href="#multiplexed-pe"><span class="std std-ref">see here</span></a>.</p></li>
<li><p>If you need to trim the primers/adapters, <a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">see here</span></a>.</p></li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Be sure that all sequences have <strong>same orientation</strong> (5’-3’ or 3’-5’) in your input data set(s)! If sequences are in <strong>mixed orientation</strong>
(i.e. some sequences are recorded as 5’-3’ and some as 3’-5’; as usually in the raw data),
then exactly the same ASV may be reported twice, where one is just the reverse complementary ASV: 1) ASV with sequence orientation of 5’-3’; and 2) ASV with sequence orientation of 3’-5’. <strong>Reorient sequences</strong> based on primer sequene using <code class="docutils literal notranslate"><span class="pre">REORIENT</span></code> panel; <a class="reference internal" href="quicktools.html#reorient"><span class="std std-ref">see here</span></a>.</p>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>When working with your own data, then please check that the paired-end data file names contain “R1” and “R2” strings
(to correctly identify the paired-end reads by PipeCraft).</p>
<div class="line-block">
<div class="line"><em>Example:</em></div>
<div class="line"><em>F3D0_S188_L001_R1_001.fastq</em></div>
<div class="line"><em>F3D0_S188_L001_R2_001.fastq</em></div>
</div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">1.</span> <span class="n">Select</span> <span class="n">working</span> <span class="n">directory</span> <span class="n">by</span> <span class="n">pressing</span> <span class="n">the</span> <span class="s1">&#39;SELECT WORKDIR&#39;</span> <span class="n">button</span><span class="o">.</span>
</pre></div>
</div>
<div class="line-block">
<div class="line">Secify</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">sequencing</span> <span class="pre">data</span> <span class="pre">format</span></code> as <strong>demultiplexed</strong>;</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">sequence</span> <span class="pre">files</span> <span class="pre">extension</span></code> as <strong>*.fastq</strong>;</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">sequencing</span> <span class="pre">read</span> <span class="pre">types</span></code> as <strong>paired-end</strong>.</div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">2.</span> <span class="n">Select</span> <span class="s1">&#39;ASVs workflow&#39;</span> <span class="n">panel</span> <span class="p">(</span><span class="n">right</span><span class="o">-</span><span class="n">ribbon</span><span class="p">)</span> <span class="ow">and</span> <span class="n">check</span> <span class="n">that</span> <span class="n">docker</span> <span class="ow">is</span> <span class="n">running</span> <span class="p">(</span><span class="n">green</span> <span class="n">icon</span><span class="p">);</span>
</pre></div>
</div>
<ul class="simple">
<li><p>Here, working with demultiplexed data, where primers have already been removed; so <strong>do not tick</strong> <code class="docutils literal notranslate"><span class="pre">DEMULTIPLEX</span></code>, <code class="docutils literal notranslate"><span class="pre">REORIENT</span></code>, <code class="docutils literal notranslate"><span class="pre">CUT</span> <span class="pre">PRIMERS</span></code>
(<a class="reference internal" href="#multiplexed-pe"><span class="std std-ref">see here</span></a> to analyse multiplexed data, and <a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">here</span></a> if you need to cut primers/adapters).</p></li>
</ul>
<p><a class="reference internal" href="_images/ASVs_tut1.png"><img alt="Alternative text" src="_images/ASVs_tut1.png" style="width: 1000px;" /></a></p>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<div class="highlight-default notranslate" id="rest-of-pe-asv-workflow"><div class="highlight"><pre><span></span><span class="mf">3.</span> <span class="s1">&#39;QUALITY FILTERING&#39;</span>
</pre></div>
</div>
<p>Before adjusting quality filtering settings, let’s have a look on the <strong>quality profile</strong> of our example data set. Below quality profile plot was generated using <code class="docutils literal notranslate"><span class="pre">QualityCheck</span></code> panel (<a class="reference internal" href="overview.html#qualitycheck"><span class="std std-ref">see here</span></a>).</p>
<p><a class="reference internal" href="_images/fastqc_per_base_sequence_quality_plot.png"><img alt="Alternative text" src="_images/fastqc_per_base_sequence_quality_plot.png" style="width: 850px;" /></a></p>
<p>In this case, all <strong>R1 files are represented with green lines</strong>, indicating good average quality per file. However, all <strong>R2 files are either yellow or red</strong>, indicating a drop in quality scores.
Lower qualities of R2 reads are characteristic for Illumina sequencing data, and is not too alarming. DADA2 algoritm is robust to lower quality sequences, but removing the low quality read parts
will improve the DADA2 sensitivity to rare sequence variants.</p>
<ul class="simple">
<li><p><strong>Click on</strong> <code class="docutils literal notranslate"><span class="pre">QUALITY</span> <span class="pre">FILTERING</span></code> <strong>to expand the panel</strong></p></li>
<li><p>specify identifier strings for <code class="docutils literal notranslate"><span class="pre">read</span> <span class="pre">R1</span></code> and <code class="docutils literal notranslate"><span class="pre">read</span> <span class="pre">R2</span></code>. Here, fastq file names = F3D0_S188_L001_R1_001.fastq, F3D0_S188_L001_R2_001.fastq etc…; <strong>_R1</strong> and <strong>_R2</strong> are common identifiers for all files.</p></li>
<li><p>specify <code class="docutils literal notranslate"><span class="pre">samp</span> <span class="pre">ID</span></code> (sample identifier). Here <strong>_</strong> (underscore), which denotes that sample name is a string before the first <strong>_</strong> in the fastq file name.</p></li>
<li><p>trim reads to specified length to remove low quality ends. Set <code class="docutils literal notranslate"><span class="pre">truncLen</span></code> to 240 for trimming R1 reads and <code class="docutils literal notranslate"><span class="pre">truncLen</span> <span class="pre">R2</span></code> to 160 to trim R2 reads. Latter positions represent the approximate positions where sequence quality drps notably
(quality profile figure above). Be sure to consider the amplicon length before applying <code class="docutils literal notranslate"><span class="pre">truncLen</span></code> options, so that R1 and R2 reads would still overlap for the <code class="docutils literal notranslate"><span class="pre">MERGE</span> <span class="pre">PAIRS</span></code> process.</p></li>
<li><p>other settings as default.</p></li>
</ul>
<p><em>(click on the image for enlargement)</em>
<a class="reference internal" href="_images/DADA2_quality_filt_expand_dadaTut.png"><img alt="Alternative text" src="_images/DADA2_quality_filt_expand_dadaTut.png" style="width: 550px;" /></a></p>
<div class="line-block">
<div class="line"><em>This step performs quality filtering.</em></div>
<div class="line"><em>Quality filtering settings</em> <a class="reference internal" href="pre-defined_pipelines.html#dada2-qual-filt"><span class="std std-ref">here</span></a></div>
<div class="line"><br /></div>
<div class="line"><strong>Output</strong> directory = <code class="docutils literal notranslate"><span class="pre">qualFiltered_out</span></code>:</div>
<div class="line">*_filt.fastq          = quality filtered sequences per sample in FASTQ format</div>
<div class="line">seq_count_summary.txt = summary of sequence counts per sample</div>
<div class="line">FASTA/*_filt.fasta    = quality filtered sequences per sample in FASTA format</div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">4.</span> <span class="n">Here</span><span class="p">,</span> <span class="n">we</span> <span class="n">use</span> <span class="n">default</span> <span class="s1">&#39;DENOISE&#39;</span> <span class="ow">and</span> <span class="s1">&#39;MERGE PAIRS&#39;</span> <span class="n">settings</span>
</pre></div>
</div>
<div class="line-block">
<div class="line"><em>This step performs denoising and merging of paired-end sequences.</em></div>
<div class="line"><em>Denoise settings</em> : <a class="reference internal" href="pre-defined_pipelines.html#dada2-denoise"><span class="std std-ref">here</span></a>, <em>merge pairs settings</em> <a class="reference internal" href="pre-defined_pipelines.html#dada2-merge-pairs"><span class="std std-ref">here</span></a>)</div>
<div class="line"><br /></div>
<div class="line"><strong>Output</strong> directory = <code class="docutils literal notranslate"><span class="pre">denoised_assembled.dada2</span></code>.</div>
<div class="line">*.merged_ASVs.fasta   = denoised and assembled ASVs per sample. ‘Size’ denotes the abundance of the ASV sequence</div>
<div class="line">Error_rates_R1.pdf    = plots for estimated R1 error rates</div>
<div class="line">Error_rates_R2.pdf    = plots for estimated R2 error rates</div>
<div class="line">seq_count_summary.txt = summary of sequence and ASV counts per sample</div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">5.</span> <span class="n">Default</span> <span class="n">settings</span> <span class="k">for</span> <span class="s1">&#39;CHIMERA FILTERING&#39;</span>
</pre></div>
</div>
<p>(method = consensus)</p>
<div class="line-block">
<div class="line"><em>This step performs chimera filtering on denoised and merged reads.</em></div>
<div class="line"><em>ASV table is generated during this step</em></div>
<div class="line"><em>Chimera filtering settings</em> <a class="reference internal" href="pre-defined_pipelines.html#dada2-chimeras"><span class="std std-ref">here</span></a></div>
<div class="line"><br /></div>
<div class="line"><strong>Output</strong> directories -&gt;</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">chimeraFiltered_out.dada2</span></code>:</div>
<div class="line">*.chimFilt_ASVs.fasta = chimera filtered ASVs per sample. ‘Size’ denotes the abundance of the ASV sequence.</div>
<div class="line">seq_count_summary.txt = summary of sequence and ASV counts per sample</div>
<div class="line">*.chimeras.fasta      = ASVs per sample that were flagged as chimeras (and thus discarded)</div>
</div>
<div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">ASVs_out.dada2</span></code>:</div>
<div class="line">ASVs_table.txt = ASV distribution table per sample (tab delimited file)</div>
<div class="line">ASVs.fasta     = FASTA formatted representative ASV sequences (this file is used for taxonomy assignment)</div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">6.</span> <span class="s1">&#39;ASSIGN TAXONOMY&#39;</span>
</pre></div>
</div>
<ul class="simple">
<li><p>Click on ‘ASSIGN TAXONOMY’ to expand the panel</p></li>
<li><p>press <code class="docutils literal notranslate"><span class="pre">DOWNLOAD</span> <span class="pre">DATABASES</span></code> which direct you to the DADA2-formatted reference databases <a class="reference external" href="https://benjjneb.github.io/dada2/training.html">web page</a>.</p></li>
<li><p>download SILVA (silva_nr99_v138.1_wSpecies_train_set.fa.gz) database for assigning taxonomy to our 16S ASVs. <a class="reference external" href="https://zenodo.org/record/4587955/files/silva_nr99_v138.1_wSpecies_train_set.fa.gz?download=1">Download link here</a></p></li>
<li><p>specify the location of your downloaded DADA2 database by pressing <code class="docutils literal notranslate"><span class="pre">SELECT</span> <span class="pre">FASTA</span></code></p></li>
<li><p>since primers were already removed from this data set, we could not <a class="reference internal" href="quicktools.html#reorient"><span class="std std-ref">reorient all sequences to uniform orientation as based on primers</span></a>. Therefore, <strong>swithc ON</strong> <code class="docutils literal notranslate"><span class="pre">tryRC</span></code>
to include reverse-complement search.</p></li>
</ul>
<p><a class="reference internal" href="_images/DADA2_assign_taxRC.png"><img alt="Alternative text" src="_images/DADA2_assign_taxRC.png" style="width: 550px;" /></a></p>
<div class="line-block">
<div class="line"><em>This step assigns taxonomy to ASVs using DADA2</em> <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/manuals/dada2/man/dada2.pdf">assignTaxonomy</a> function.</div>
<div class="line"><em>Assign taxonomy settings</em> <a class="reference internal" href="pre-defined_pipelines.html#dada2-taxonomy"><span class="std std-ref">here</span></a></div>
<div class="line"><br /></div>
<div class="line"><strong>Output</strong> directory = <code class="docutils literal notranslate"><span class="pre">taxonomy_out.dada2</span></code>:</div>
<div class="line">taxonomy.txt = classifier results with bootstrap values</div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>6.1. Save the workflow by pressing ``SAVE WORKFLOW`` button on the right-ribbon.
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">7.</span>  <span class="n">Press</span><span class="o">**</span> <span class="s1">&#39;START&#39;</span> <span class="o">**</span><span class="n">to</span> <span class="n">start</span> <span class="n">the</span> <span class="n">analyses</span><span class="o">.</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>When running the panel for the first time, a docker image will be pulled first to start the process.</p>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">When</span> <span class="n">done</span><span class="p">,</span> <span class="s1">&#39;workflow finished&#39;</span> <span class="n">window</span> <span class="n">will</span> <span class="n">be</span> <span class="n">displayed</span><span class="o">.</span>
</pre></div>
</div>
<p><a class="reference internal" href="_images/workflow_finished.png"><img alt="Alternative text" src="_images/workflow_finished.png" style="width: 300px;" /></a></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<dl class="simple">
<dt>Press <code class="docutils literal notranslate"><span class="pre">STOP</span> <span class="pre">WORKFLOW</span></code> to stop.</dt><dd><p><a class="reference internal" href="_images/stop_workflow.png"><img alt="Alternative text" src="_images/stop_workflow.png" style="width: 200px;" /></a></p>
</dd>
</dl>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p>-&gt;</p>
<section id="examine-the-outputs">
<h4><a class="toc-backref" href="#id8">Examine the outputs</a><a class="headerlink" href="#examine-the-outputs" title="Permalink to this headline"></a></h4>
<p>Several process-specific output folders are generated:</p>
<div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">qualFiltered_out</span></code> -&gt; quality filtered paired-end <strong>fastq</strong> files per sample</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">denoised_assembled.dada2</span></code> -&gt; denoised and assembled <strong>fasta</strong> files per sample (and error rate plots)</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">chimeraFiltered_out.dada2</span></code> –&gt; chimera filtered <strong>fasta</strong> files per sample</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">ASVs_out.dada2</span></code> –&gt; <strong>ASVs table</strong> (ASVs_table.txt), and ASV sequences (ASVs.fasta) file</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">taxonomy_out.dada2</span></code>–&gt; ASVs <strong>taxonomy table</strong> (taxonomy.txt)</div>
</div>
<p id="seq-count-summary">Each folder (except ASVs_out.dada2 and taxonomy_out.dada2) contain
<strong>summary of the sequence counts</strong> (seq_count_summary.txt).
Examine those to track the read counts throughout the pipeline.</p>
<blockquote>
<div><p>For example, merging the seq_count_summary.txt file in <code class="docutils literal notranslate"><span class="pre">qualFiltered_out</span></code> with the seq_count_summary.txt file from <code class="docutils literal notranslate"><span class="pre">chimeraFiltered_out.dada2</span></code> forms a table for examining sequence counts throughout the
pipeline and number of ASVs per sample.</p>
</div></blockquote>
<table class="docutils align-default">
<colgroup>
<col style="width: 13%" />
<col style="width: 9%" />
<col style="width: 21%" />
<col style="width: 11%" />
<col style="width: 29%" />
<col style="width: 18%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>sample</p></th>
<th class="head"><p>input</p></th>
<th class="head"><p>qualFiltered</p></th>
<th class="head"><p>merged</p></th>
<th class="head"><p>chimeraFiltered</p></th>
<th class="head"><p>no.of ASVs</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>F3D0</p></td>
<td><p>7793</p></td>
<td><p>7113</p></td>
<td><p>6540</p></td>
<td><p>6528</p></td>
<td><p>106</p></td>
</tr>
<tr class="row-odd"><td><p>F3D141</p></td>
<td><p>5958</p></td>
<td><p>5463</p></td>
<td><p>4986</p></td>
<td><p>4863</p></td>
<td><p>74</p></td>
</tr>
<tr class="row-even"><td><p>F3D142</p></td>
<td><p>3183</p></td>
<td><p>2914</p></td>
<td><p>2595</p></td>
<td><p>2521</p></td>
<td><p>48</p></td>
</tr>
<tr class="row-odd"><td><p>F3D143</p></td>
<td><p>3178</p></td>
<td><p>2941</p></td>
<td><p>2552</p></td>
<td><p>2518</p></td>
<td><p>56</p></td>
</tr>
<tr class="row-even"><td><p>F3D144</p></td>
<td><p>4827</p></td>
<td><p>4312</p></td>
<td><p>3627</p></td>
<td><p>3488</p></td>
<td><p>47</p></td>
</tr>
<tr class="row-odd"><td><p>F3D145</p></td>
<td><p>7377</p></td>
<td><p>6741</p></td>
<td><p>6079</p></td>
<td><p>5820</p></td>
<td><p>72</p></td>
</tr>
<tr class="row-even"><td><p>F3D146</p></td>
<td><p>5021</p></td>
<td><p>4560</p></td>
<td><p>3968</p></td>
<td><p>3879</p></td>
<td><p>84</p></td>
</tr>
<tr class="row-odd"><td><p>F3D147</p></td>
<td><p>17070</p></td>
<td><p>15637</p></td>
<td><p>14231</p></td>
<td><p>13006</p></td>
<td><p>103</p></td>
</tr>
<tr class="row-even"><td><p>F3D148</p></td>
<td><p>12405</p></td>
<td><p>11413</p></td>
<td><p>10529</p></td>
<td><p>9935</p></td>
<td><p>97</p></td>
</tr>
<tr class="row-odd"><td><p>F3D149</p></td>
<td><p>13083</p></td>
<td><p>12017</p></td>
<td><p>11154</p></td>
<td><p>10653</p></td>
<td><p>112</p></td>
</tr>
<tr class="row-even"><td><p>F3D150</p></td>
<td><p>5509</p></td>
<td><p>5032</p></td>
<td><p>4349</p></td>
<td><p>4240</p></td>
<td><p>78</p></td>
</tr>
<tr class="row-odd"><td><p>F3D1</p></td>
<td><p>5869</p></td>
<td><p>5299</p></td>
<td><p>5028</p></td>
<td><p>5017</p></td>
<td><p>100</p></td>
</tr>
<tr class="row-even"><td><p>F3D2</p></td>
<td><p>19620</p></td>
<td><p>18075</p></td>
<td><p>17431</p></td>
<td><p>16835</p></td>
<td><p>134</p></td>
</tr>
<tr class="row-odd"><td><p>F3D3</p></td>
<td><p>6758</p></td>
<td><p>6250</p></td>
<td><p>5853</p></td>
<td><p>5491</p></td>
<td><p>68</p></td>
</tr>
<tr class="row-even"><td><p>F3D5</p></td>
<td><p>4448</p></td>
<td><p>4052</p></td>
<td><p>3716</p></td>
<td><p>3716</p></td>
<td><p>86</p></td>
</tr>
<tr class="row-odd"><td><p>F3D6</p></td>
<td><p>7989</p></td>
<td><p>7369</p></td>
<td><p>6865</p></td>
<td><p>6679</p></td>
<td><p>90</p></td>
</tr>
<tr class="row-even"><td><p>F3D7</p></td>
<td><p>5129</p></td>
<td><p>4765</p></td>
<td><p>4428</p></td>
<td><p>4217</p></td>
<td><p>61</p></td>
</tr>
<tr class="row-odd"><td><p>F3D8</p></td>
<td><p>5294</p></td>
<td><p>4871</p></td>
<td><p>4576</p></td>
<td><p>4547</p></td>
<td><p>99</p></td>
</tr>
<tr class="row-even"><td><p>F3D9</p></td>
<td><p>7070</p></td>
<td><p>6504</p></td>
<td><p>6092</p></td>
<td><p>6015</p></td>
<td><p>106</p></td>
</tr>
<tr class="row-odd"><td><p>Mock</p></td>
<td><p>4779</p></td>
<td><p>4314</p></td>
<td><p>4269</p></td>
<td><p>4269</p></td>
<td><p>20</p></td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">ASVs_out.dada2</span></code> directory contains <strong>ASVs table</strong> (ASVs_table.txt), where the <strong>1st column</strong> represents ASV identifiers,
<strong>2nd column</strong> representative sequences of ASVs,
and all following columns represent samples (number of sequences per ASV in a sample). This is tab delimited text file.</p>
<p><em>ASVs_table.txt; first 4 samples</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 12%" />
<col style="width: 33%" />
<col style="width: 12%" />
<col style="width: 14%" />
<col style="width: 14%" />
<col style="width: 14%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>ASV</p></th>
<th class="head"><p>Sequence</p></th>
<th class="head"><p>F3D0</p></th>
<th class="head"><p>F3D141</p></th>
<th class="head"><p>F3D142</p></th>
<th class="head"><p>F3D143</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>ASV_1</p></td>
<td><p>TACGGAGGATG…</p></td>
<td><p>579</p></td>
<td><p>444</p></td>
<td><p>289</p></td>
<td><p>228</p></td>
</tr>
<tr class="row-odd"><td><p>ASV_2</p></td>
<td><p>TACGGAGGATG…</p></td>
<td><p>345</p></td>
<td><p>362</p></td>
<td><p>304</p></td>
<td><p>176</p></td>
</tr>
<tr class="row-even"><td><p>ASV_3</p></td>
<td><p>TACGGAGGATG…</p></td>
<td><p>449</p></td>
<td><p>345</p></td>
<td><p>158</p></td>
<td><p>204</p></td>
</tr>
<tr class="row-odd"><td><p>ASV_4</p></td>
<td><p>TACGGAGGATG…</p></td>
<td><p>430</p></td>
<td><p>502</p></td>
<td><p>164</p></td>
<td><p>231</p></td>
</tr>
<tr class="row-even"><td><p>ASV_5</p></td>
<td><p>TACGGAGGATC…</p></td>
<td><p>154</p></td>
<td><p>189</p></td>
<td><p>180</p></td>
<td><p>130</p></td>
</tr>
<tr class="row-odd"><td><p>ASV_6</p></td>
<td><p>TACGGAGGATG…</p></td>
<td><p>470</p></td>
<td><p>331</p></td>
<td><p>181</p></td>
<td><p>244</p></td>
</tr>
<tr class="row-even"><td><p>ASV_7</p></td>
<td><p>TACGGAGGATG…</p></td>
<td><p>282</p></td>
<td><p>243</p></td>
<td><p>163</p></td>
<td><p>152</p></td>
</tr>
<tr class="row-odd"><td><p>ASV_8</p></td>
<td><p>TACGGAGGATT…</p></td>
<td><p>184</p></td>
<td><p>321</p></td>
<td><p>89</p></td>
<td><p>83</p></td>
</tr>
<tr class="row-even"><td><p>ASV_9</p></td>
<td><p>TACGGAGGATG…</p></td>
<td><p>45</p></td>
<td><p>167</p></td>
<td><p>89</p></td>
<td><p>109</p></td>
</tr>
</tbody>
</table>
<p>The <strong>ASV sequences</strong> are representad also in the fasta file (ASVs.fasta) in <code class="docutils literal notranslate"><span class="pre">ASVs_out.dada2</span></code> directory.</p>
<p>Result from the taxonomy annotation process - <strong>taxonomy table</strong> (taxonomy.txt) - is located at the <code class="docutils literal notranslate"><span class="pre">taxonomy_out.dada2</span></code> directory.
“NA” denotes that the ASV was not assigned to corresponding taxonomic unit.
Last columns with integers (for ‘Kingdom’ to ‘Species’) represent bootstrap values for the correspoinding taxonomic unit.</p>
<p><em>taxonomy.txt; first 10 ASVs</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 4%" />
<col style="width: 6%" />
<col style="width: 5%" />
<col style="width: 8%" />
<col style="width: 7%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 19%" />
<col style="width: 6%" />
<col style="width: 4%" />
<col style="width: 4%" />
<col style="width: 3%" />
<col style="width: 3%" />
<col style="width: 4%" />
<col style="width: 3%" />
<col style="width: 4%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>ASV</p></th>
<th class="head"><p>Sequence</p></th>
<th class="head"><p>Kingdom</p></th>
<th class="head"><p>Phylum</p></th>
<th class="head"><p>Class</p></th>
<th class="head"><p>Order</p></th>
<th class="head"><p>Family</p></th>
<th class="head"><p>Genus</p></th>
<th class="head"><p>Species</p></th>
<th class="head"><p>Kingdom</p></th>
<th class="head"><p>Phylum</p></th>
<th class="head"><p>Class</p></th>
<th class="head"><p>Order</p></th>
<th class="head"><p>Family</p></th>
<th class="head"><p>Genus</p></th>
<th class="head"><p>Species</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>ASV_1</p></td>
<td><p>TACGGAG…</p></td>
<td><p>Bacteria</p></td>
<td><p>Bacteroidota</p></td>
<td><p>Bacteroidia</p></td>
<td><p>Bacteroidales</p></td>
<td><p>Muribaculaceae</p></td>
<td><p>NA</p></td>
<td><p>NA</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
</tr>
<tr class="row-odd"><td><p>ASV_2</p></td>
<td><p>TACGGAG…</p></td>
<td><p>Bacteria</p></td>
<td><p>Bacteroidota</p></td>
<td><p>Bacteroidia</p></td>
<td><p>Bacteroidales</p></td>
<td><p>Muribaculaceae</p></td>
<td><p>NA</p></td>
<td><p>NA</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
</tr>
<tr class="row-even"><td><p>ASV_3</p></td>
<td><p>TACGGAG…</p></td>
<td><p>Bacteria</p></td>
<td><p>Bacteroidota</p></td>
<td><p>Bacteroidia</p></td>
<td><p>Bacteroidales</p></td>
<td><p>Muribaculaceae</p></td>
<td><p>NA</p></td>
<td><p>NA</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
</tr>
<tr class="row-odd"><td><p>ASV_4</p></td>
<td><p>TACGGAG…</p></td>
<td><p>Bacteria</p></td>
<td><p>Bacteroidota</p></td>
<td><p>Bacteroidia</p></td>
<td><p>Bacteroidales</p></td>
<td><p>Rikenellaceae</p></td>
<td><p>Alistipes</p></td>
<td><p>NA</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
</tr>
<tr class="row-even"><td><p>ASV_5</p></td>
<td><p>TACGGAG…</p></td>
<td><p>Bacteria</p></td>
<td><p>Bacteroidota</p></td>
<td><p>Bacteroidia</p></td>
<td><p>Bacteroidales</p></td>
<td><p>Muribaculaceae</p></td>
<td><p>NA</p></td>
<td><p>NA</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
</tr>
<tr class="row-odd"><td><p>ASV_6</p></td>
<td><p>TACGGAG…</p></td>
<td><p>Bacteria</p></td>
<td><p>Bacteroidota</p></td>
<td><p>Bacteroidia</p></td>
<td><p>Bacteroidales</p></td>
<td><p>Muribaculaceae</p></td>
<td><p>NA</p></td>
<td><p>NA</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>95</p></td>
<td><p>95</p></td>
</tr>
<tr class="row-even"><td><p>ASV_7</p></td>
<td><p>TACGTAG…</p></td>
<td><p>Bacteria</p></td>
<td><p>Firmicutes</p></td>
<td><p>Clostridia</p></td>
<td><p>Lachnospirales</p></td>
<td><p>Lachnospiraceae</p></td>
<td><p>Lachnospiraceae NK4A136 group</p></td>
<td><p>NA</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>99</p></td>
</tr>
<tr class="row-odd"><td><p>ASV_8</p></td>
<td><p>TACGGAG…</p></td>
<td><p>Bacteria</p></td>
<td><p>Bacteroidota</p></td>
<td><p>Bacteroidia</p></td>
<td><p>Bacteroidales</p></td>
<td><p>Muribaculaceae</p></td>
<td><p>NA</p></td>
<td><p>NA</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
</tr>
<tr class="row-even"><td><p>ASV_9</p></td>
<td><p>TACGGAG…</p></td>
<td><p>Bacteria</p></td>
<td><p>Bacteroidota</p></td>
<td><p>Bacteroidia</p></td>
<td><p>Bacteroidales</p></td>
<td><p>Bacteroidaceae</p></td>
<td><p>Bacteroides</p></td>
<td><p>caecimuris</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>77</p></td>
</tr>
<tr class="row-odd"><td><p>ASV_10</p></td>
<td><p>TACGGAG…</p></td>
<td><p>Bacteria</p></td>
<td><p>Bacteroidota</p></td>
<td><p>Bacteroidia</p></td>
<td><p>Bacteroidales</p></td>
<td><p>Muribaculaceae</p></td>
<td><p>NA</p></td>
<td><p>NA</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>99</p></td>
<td><p>99</p></td>
</tr>
</tbody>
</table>
<hr class="docutils" />
<div class="line-block">
<div class="line"><br /></div>
</div>
</section>
</section>
<section id="demultiplexed-paired-end-data-otu-workflow">
<span id="demultiplexed-pe-otu"></span><h3><a class="toc-backref" href="#id9">Demultiplexed paired-end data; OTU workflow</a><a class="headerlink" href="#demultiplexed-paired-end-data-otu-workflow" title="Permalink to this headline"></a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Built-in panel for OTU workflow with (mostly) vsearch.</p>
</div>
<p>Here, we perform example analyses of paired-end data using <a class="reference external" href="https://mothur.org/wiki/miseq_sop/">mothur MiSeq SOP example data set</a>.
<a class="reference external" href="https://mothur.s3.us-east-2.amazonaws.com/wiki/miseqsopdata.zip">Download example data set here</a> (35.1 Mb) and unzip it.
This data set represents demultiplexed set (per-sample fastq files) of 16S rRNA gene V4 amplicon sequences where sample indexes and primers have already been removed.</p>
<ul class="simple">
<li><p>If working with multiplexed data, <a class="reference internal" href="#multiplexed-pe"><span class="std std-ref">see here</span></a>.</p></li>
<li><p>If you need to trim the primers/adapters, <a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">see here</span></a>.</p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>When working with your own data, then consider <strong>reorienting</strong> reads; <a class="reference internal" href="quicktools.html#reorient"><span class="std std-ref">see here</span></a>. Although, in the OTU formation step (clustering), both sequence strands
will be compared to generate OTUs, the time for BLAST (taxonomy annotation step) can be reduced when there is no need to search reverse complementary matches.</p>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>When working with your own data, then please check that the paired-end data file names contain “R1” and “R2” strings
(to correctly identify the paired-end reads by PipeCraft).</p>
<div class="line-block">
<div class="line"><em>Example:</em></div>
<div class="line"><em>F3D0_S188_L001_R1_001.fastq</em></div>
<div class="line"><em>F3D0_S188_L001_R2_001.fastq</em></div>
</div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">1.</span> <span class="n">Select</span> <span class="n">working</span> <span class="n">directory</span> <span class="n">by</span> <span class="n">pressing</span> <span class="n">the</span> <span class="s1">&#39;SELECT WORKDIR&#39;</span> <span class="n">button</span><span class="o">.</span>
</pre></div>
</div>
<div class="line-block">
<div class="line">Secify</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">sequencing</span> <span class="pre">data</span> <span class="pre">format</span></code> as <strong>demultiplexed</strong>;</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">sequence</span> <span class="pre">files</span> <span class="pre">extension</span></code> as <strong>*.fastq</strong>;</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">sequencing</span> <span class="pre">read</span> <span class="pre">types</span></code> as <strong>paired-end</strong>.</div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">2.</span> <span class="n">Select</span> <span class="s1">&#39;OTU workflow&#39;</span> <span class="n">panel</span> <span class="p">(</span><span class="n">right</span><span class="o">-</span><span class="n">ribbon</span><span class="p">)</span> <span class="ow">and</span> <span class="n">check</span> <span class="n">that</span> <span class="n">docker</span> <span class="ow">is</span> <span class="n">running</span> <span class="p">(</span><span class="n">green</span> <span class="n">icon</span><span class="p">);</span>
</pre></div>
</div>
<ul class="simple">
<li><p>Here, working with demultiplexed data, where primers have already been removed; so <strong>do not tick</strong> <code class="docutils literal notranslate"><span class="pre">DEMULTIPLEX</span></code>, <code class="docutils literal notranslate"><span class="pre">REORIENT</span></code>, <code class="docutils literal notranslate"><span class="pre">CUT</span> <span class="pre">PRIMERS</span></code>
(<a class="reference internal" href="#multiplexed-pe"><span class="std std-ref">see here</span></a> to analyse multiplexed data, and <a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">here</span></a> if you need to cut primers/adapters).</p></li>
</ul>
<p><a class="reference internal" href="_images/OTUs_workflow.png"><img alt="Alternative text" src="_images/OTUs_workflow.png" style="width: 1000px;" /></a></p>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<p>Before proceeding, let’s have a look on the <strong>quality profile</strong> of our example data set. Below quality profile plot was generated using <code class="docutils literal notranslate"><span class="pre">QualityCheck</span></code> panel (<a class="reference internal" href="overview.html#qualitycheck"><span class="std std-ref">see here</span></a>).</p>
<p><a class="reference internal" href="_images/fastqc_per_base_sequence_quality_plot.png"><img alt="Alternative text" src="_images/fastqc_per_base_sequence_quality_plot.png" style="width: 850px;" /></a></p>
<p>In this case, all <strong>R1 files are represented with green lines</strong>, indicating good average quality per file. However, all <strong>R2 files are either yellow or red</strong>, indicating a drop in quality scores.
Lower qualities of R2 reads are characteristic for Illumina sequencing data, and is not too alarming. Nevertheless, we need to quality filter the data set.</p>
<div class="highlight-default notranslate" id="rest-of-pe-otu-workflow"><div class="highlight"><pre><span></span><span class="mf">3.</span> <span class="s1">&#39;MERGE PAIRS&#39;</span>
</pre></div>
</div>
<ul class="simple">
<li><p>Here, we use default settings.</p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If <code class="docutils literal notranslate"><span class="pre">include_only_R1</span></code> option = TRUE,
then unassembled R1 reads will be included to the set of assembled reads per sample.
This may be useful when working with e.g. ITS2 sequences, because the ITS2 region in some
taxa is too long for paired-end assembly using current short-read sequencing technology.
Therefore longer ITS2 amplicon sequences are discarded completely after the assembly process.
Thus, including also unassembled R1 reads (<code class="docutils literal notranslate"><span class="pre">include_only_R1</span></code> = TRUE), partial ITS2 sequences for
these taxa will be represented in the final output. But when using <a class="reference internal" href="quicktools.html#itsextractor"><span class="std std-ref">ITSx</span></a>
, keep <code class="docutils literal notranslate"><span class="pre">only_full</span></code> = FALSE and include <code class="docutils literal notranslate"><span class="pre">partial</span></code> = 50.
|
If include only R1 option = TRUE, then other specified options (lenght, max error rate etc.) have not been
applied to R1 reads in the ‘assembled’ file. Thus, additional quality filtering (if this was done before assembling)
should be run on the ‘assembled’ data. But in this built-in OTU workflow, the quality filtering step is anyway performed after merge pairs step.</p>
</div>
<div class="line-block">
<div class="line"><em>This step performs merging of paired-end sequences using vsearch –fastq_mergepairs.</em></div>
<div class="line"><em>Merge pairs settings</em> <a class="reference internal" href="quicktools.html#merge-pairs"><span class="std std-ref">here</span></a>)</div>
<div class="line"><br /></div>
<div class="line"><strong>Output</strong> directory = <code class="docutils literal notranslate"><span class="pre">assembled_out</span></code>.</div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">4.</span> <span class="s1">&#39;QUALITY FILTERING&#39;</span>
</pre></div>
</div>
<ul class="simple">
<li><p><strong>Click on</strong> <code class="docutils literal notranslate"><span class="pre">QUALITY</span> <span class="pre">FILTERING</span></code> <strong>to expand the panel</strong></p></li>
<li><p>specify <code class="docutils literal notranslate"><span class="pre">maxee</span></code> (maximum number of expected errors per sequence), here we use 1 (<a class="reference external" href="https://drive5.com/usearch/manual/exp_errs.html">see here what is maxee</a>).</p></li>
<li><p>specify <code class="docutils literal notranslate"><span class="pre">maxNs</span></code> (maximum number of Ns in the sequences). Here, we will discard any sequence that contains N (ambiguously recorded nucleotide) by setting the value to 0.</p></li>
<li><p>other settings as default.</p></li>
</ul>
<p><a class="reference internal" href="_images/vsearch_qfilt.png"><img alt="Alternative text" src="_images/vsearch_qfilt.png" style="width: 600px;" /></a></p>
<div class="line-block">
<div class="line"><em>This step performs quality filtering using vsearch.</em></div>
<div class="line"><em>vsearch quality filtering settings</em> <a class="reference internal" href="quicktools.html#qfilt-vsearch"><span class="std std-ref">here</span></a></div>
<div class="line"><br /></div>
<div class="line"><strong>Output</strong> directory = <code class="docutils literal notranslate"><span class="pre">qualFiltered_out</span></code></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">5.</span> <span class="s1">&#39;CHIMERA FILTERING&#39;</span>
</pre></div>
</div>
<ul class="simple">
<li><p><strong>Click on</strong> <code class="docutils literal notranslate"><span class="pre">CHIMERA</span> <span class="pre">FILTERING</span></code> <strong>to expand the panel</strong></p></li>
<li><p>specify <code class="docutils literal notranslate"><span class="pre">pre</span> <span class="pre">cluster</span></code> threshold as 0.97 (that is 97%; when planning to use 97% sequence similarity threshold also for clustering reads into OTUs).</p></li>
<li><p>here, we perform only <code class="docutils literal notranslate"><span class="pre">denovo</span></code> chimera filtering</p></li>
<li><p>other settings as default.</p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Tick <code class="docutils literal notranslate"><span class="pre">reference</span> <span class="pre">based</span></code> if there is appropriate database for reference based chimera filtering
(such as e.g. <a class="reference external" href="https://unite.ut.ee/sh_files/uchime_reference_dataset_28.06.2017.zip">UNITE for ITS reads</a>).</p>
</div>
<p><a class="reference internal" href="_images/vsearch_chimeraFilt.png"><img alt="Alternative text" src="_images/vsearch_chimeraFilt.png" style="width: 600px;" /></a></p>
<div class="line-block">
<div class="line"><em>This step performs chimera filtering using vsearch</em></div>
<div class="line"><em>Chimera filtering settings</em> <a class="reference internal" href="quicktools.html#chimfilt"><span class="std std-ref">here</span></a></div>
<div class="line"><br /></div>
<div class="line"><strong>Output</strong> directory = <code class="docutils literal notranslate"><span class="pre">chimeraFiltered_out</span></code></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">6.</span> <span class="n">Consideration</span> <span class="n">when</span> <span class="n">working</span> <span class="k">with</span> <span class="n">ITS</span> <span class="n">data</span>
</pre></div>
</div>
<p>Identify and extract the ITS regions using ITSx; <a class="reference internal" href="quicktools.html#itsextractor"><span class="std std-ref">see here</span></a></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>because ITSx outputs multiple directories for different ITS sub-regions
<code class="docutils literal notranslate"><span class="pre">CLUSTERING</span></code> and <code class="docutils literal notranslate"><span class="pre">ASSIGN</span> <span class="pre">TAXONOMY</span></code> will be disabled after ‘ITS EXTRACTOR’.
Select appropriate ITSx output folder for CLUSTERING after the process is finished
[‘ADD STEP’ -&gt; <code class="docutils literal notranslate"><span class="pre">CLUSTERING</span></code> (vsearch)].</p>
</div>
<div class="line-block">
<div class="line"><em>This step extracts ITS reads using ITSx</em></div>
<div class="line"><em>ITSx settings</em> <a class="reference internal" href="quicktools.html#itsextractor"><span class="std std-ref">here</span></a></div>
<div class="line"><br /></div>
<div class="line"><strong>Output</strong> directory = <code class="docutils literal notranslate"><span class="pre">ITSx_out</span></code></div>
<div class="line"><br /></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">7.</span> <span class="s1">&#39;CLUSTERING&#39;</span>
</pre></div>
</div>
<ul class="simple">
<li><p>Here, we use default settings by clustering the reads using 97% similarity threshold</p></li>
</ul>
<div class="line-block">
<div class="line"><em>This step performs clustering using vsearch.</em></div>
<div class="line"><em>vsearch clustering settings</em> <a class="reference internal" href="quicktools.html#clustering"><span class="std std-ref">here</span></a></div>
<div class="line"><br /></div>
<div class="line"><strong>Output</strong> directory = <code class="docutils literal notranslate"><span class="pre">clustering_out</span></code></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">8.</span> <span class="s1">&#39;ASSIGN TAXONOMY&#39;</span>
</pre></div>
</div>
<ul class="simple">
<li><p>Tick <code class="docutils literal notranslate"><span class="pre">ASSIGN</span> <span class="pre">TAXONOMY</span></code> to perform taxonomy assignment with BLAST</p></li>
<li><p>download SILVA 99% database <a class="reference internal" href="quicktools.html#databases"><span class="std std-ref">here</span></a> (SILVA_138.1_SSURef_NR99_tax_silva.fasta.gz)</p></li>
<li><p><strong>unzip</strong> the downloaded database and place this into separete folder (to automatically make blast database from that fasta file)</p></li>
<li><p>specify the location of your downloaded SILVA database by pressing <code class="docutils literal notranslate"><span class="pre">SELECT</span> <span class="pre">FILE</span></code> under ‘database file’ option</p></li>
<li><p>since primers were already removed from this data set, we could not <a class="reference internal" href="quicktools.html#reorient"><span class="std std-ref">reorient all sequences to uniform orientation as based on primers</span></a>. Therefore, <strong>keep ON</strong> the <code class="docutils literal notranslate"><span class="pre">strands</span></code> = both to include reverse-complement search.</p></li>
</ul>
<p><a class="reference internal" href="_images/assign_taxonomy_blast.png"><img alt="Alternative text" src="_images/assign_taxonomy_blast.png" style="width: 600px;" /></a></p>
<div class="line-block">
<div class="line"><em>This step assigns taxonomy to OTUs using BLAST</em></div>
<div class="line"><em>Assign taxonomy settings</em> <a class="reference internal" href="quicktools.html#assign-taxonomy"><span class="std std-ref">here</span></a></div>
<div class="line"><br /></div>
<div class="line"><strong>Output</strong> directory = <code class="docutils literal notranslate"><span class="pre">taxonomy_out</span></code></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>8.1. Save the workflow by pressing ``SAVE WORKFLOW`` button on the right-ribbon.
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">1.</span>  <span class="n">Press</span><span class="o">**</span> <span class="s1">&#39;START&#39;</span> <span class="o">**</span><span class="n">to</span> <span class="n">start</span> <span class="n">the</span> <span class="n">analyses</span><span class="o">.</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>When running the panel for the first time, a docker image will be pulled first to start the process.</p>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">When</span> <span class="n">done</span><span class="p">,</span> <span class="s1">&#39;workflow finished&#39;</span> <span class="n">window</span> <span class="n">will</span> <span class="n">be</span> <span class="n">displayed</span><span class="o">.</span>
</pre></div>
</div>
<p><a class="reference internal" href="_images/workflow_finished.png"><img alt="Alternative text" src="_images/workflow_finished.png" style="width: 300px;" /></a></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<dl class="simple">
<dt>Press <code class="docutils literal notranslate"><span class="pre">STOP</span> <span class="pre">WORKFLOW</span></code> to stop.</dt><dd><p><a class="reference internal" href="_images/stop_workflow.png"><img alt="Alternative text" src="_images/stop_workflow.png" style="width: 200px;" /></a></p>
</dd>
</dl>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p>-&gt;</p>
<section id="id3">
<h4><a class="toc-backref" href="#id10">Examine the outputs</a><a class="headerlink" href="#id3" title="Permalink to this headline"></a></h4>
<p>Several process-specific output folders are generated:</p>
<div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">assembled_out</span></code> –&gt; assembled <strong>fastq</strong> files per sample</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">qualFiltered_out</span></code> –&gt; quality filtered <strong>fastq</strong> files per sample</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">chimeraFiltered_out</span></code> –&gt; chimera filtered <strong>fasta</strong> files per sample</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">clustering_out</span></code> –&gt; <strong>OTU table</strong> (OTU_table.txt), and OTU sequences (OTUs.fasta) file</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">taxonomy_out</span></code>–&gt; BLAST hits for the OTUs (BLAST_1st_best_hit.txt and BLAST_10_best_hits.txt)</div>
</div>
<p>Each folder (except clustering_out and taxonomy_out) contain
<strong>summary of the sequence counts</strong> (seq_count_summary.txt).
Examine those to track the read counts throughout the pipeline (<a class="reference internal" href="#seq-count-summary"><span class="std std-ref">example here</span></a>)</p>
<p><code class="docutils literal notranslate"><span class="pre">clustering_out</span></code> directory contains <strong>OTUs table</strong> (OTUs_table.txt), where the <strong>1st column</strong> represents OTU identifiers,
and all following columns represent samples (number of sequences per OTU in a sample).
The <strong>OTU sequences</strong> are representad in the fasta file (OTUs.fasta) in <code class="docutils literal notranslate"><span class="pre">clustering_out</span></code> directory.</p>
<p><em>OTUs_table.txt; first 4 samples</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 42%" />
<col style="width: 15%" />
<col style="width: 15%" />
<col style="width: 15%" />
<col style="width: 15%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>OTU_id</p></th>
<th class="head"><p>F3D0_S188_L001</p></th>
<th class="head"><p>F3D1_S189_L001</p></th>
<th class="head"><p>F3D2_S190_L001</p></th>
<th class="head"><p>F3D3_S191_L001</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>00fc1569196587dde0462c7d806cc05774f61bfa</p></td>
<td><p>106</p></td>
<td><p>271</p></td>
<td><p>584</p></td>
<td><p>20</p></td>
</tr>
<tr class="row-odd"><td><p>02d84ed0175c2c79e8379a99cffb6dbc7f6a6bd9</p></td>
<td><p>81</p></td>
<td><p>44</p></td>
<td><p>88</p></td>
<td><p>14</p></td>
</tr>
<tr class="row-even"><td><p>0407ee3bd15ca7206a75d02bb41732516adaaa88</p></td>
<td><p>3</p></td>
<td><p>4</p></td>
<td><p>3</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><p>042e5f0b5e38dff09f7ad58b6849fb17ec5503b9</p></td>
<td><p>20</p></td>
<td><p>83</p></td>
<td><p>131</p></td>
<td><p>4</p></td>
</tr>
<tr class="row-even"><td><p>07411b848fcda497fd29944d351b8a2ec7dc2bd4</p></td>
<td><p>1</p></td>
<td><p>0</p></td>
<td><p>2</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><p>07e7806a732c67ef090b6b279b74a87fefad9e8e</p></td>
<td><p>18</p></td>
<td><p>22</p></td>
<td><p>83</p></td>
<td><p>7</p></td>
</tr>
<tr class="row-even"><td><p>0836d270877aed22cd247f7e703b9247fb339127</p></td>
<td><p>1</p></td>
<td><p>1</p></td>
<td><p>0</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><p>0aa6e7da5819c11973f186cb35b3f4f58275fb04</p></td>
<td><p>1</p></td>
<td><p>4</p></td>
<td><p>5</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-even"><td><p>0c1c219a4756bb729e5f0ceb7d82d932bbfa0c5e</p></td>
<td><p>18</p></td>
<td><p>17</p></td>
<td><p>40</p></td>
<td><p>7</p></td>
</tr>
</tbody>
</table>
<p>Results from the taxonomy annotation process (BLAST) are located at the <code class="docutils literal notranslate"><span class="pre">taxonomy_out</span></code> directory (BLAST_1st_best_hit.txt and BLAST_10_best_hits.txt).
<strong>Blast values are separated by</strong> <code class="docutils literal notranslate"><span class="pre">+</span></code> and <code class="docutils literal notranslate"><span class="pre">tab</span></code> [be sure to specify the delimiter when aligning columns in e.g. LibreOffice or Excel].
“NO_BLAST_HIT” denotes that the OTU sequence did not get any match againt the selected database.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 16%" />
<col style="width: 84%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>blast values</p></th>
<th class="head"></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>score</p></td>
<td><p>blast score</p></td>
</tr>
<tr class="row-odd"><td><p>e-value</p></td>
<td><p>blast e-value</p></td>
</tr>
<tr class="row-even"><td><p>query len</p></td>
<td><p>query (i.e. OTU/ASV) sequence length</p></td>
</tr>
<tr class="row-odd"><td><p>query start</p></td>
<td><p>start position of match in the query seq</p></td>
</tr>
<tr class="row-even"><td><p>query end</p></td>
<td><p>end position of match in the query seq</p></td>
</tr>
<tr class="row-odd"><td><p>target len</p></td>
<td><p>target seq length in the database</p></td>
</tr>
<tr class="row-even"><td><p>target start</p></td>
<td><p>start position of match in the target seq</p></td>
</tr>
<tr class="row-odd"><td><p>target end</p></td>
<td><p>end position of match in the target seq</p></td>
</tr>
<tr class="row-even"><td><p>align len</p></td>
<td><p>alignment length of query and target</p></td>
</tr>
<tr class="row-odd"><td><p>identities</p></td>
<td><p>number of identical matches</p></td>
</tr>
<tr class="row-even"><td><p>gaps</p></td>
<td><p>number of gaps in the alignment</p></td>
</tr>
<tr class="row-odd"><td><p>coverage</p></td>
<td><div class="line-block">
<div class="line">query coverage percentage against the target sequence</div>
<div class="line">(100 percent is full-length match;</div>
<div class="line">low coverage may indicate presence of <strong>chimeric</strong> sequence/OTU/ASV)</div>
</div>
</td>
</tr>
<tr class="row-even"><td><p>id</p></td>
<td><p>identity percentage against the target sequence</p></td>
</tr>
</tbody>
</table>
<hr class="docutils" />
<div class="line-block">
<div class="line"><br /></div>
</div>
</section>
</section>
<section id="multiplexed-library">
<span id="multiplexed-pe"></span><h3><a class="toc-backref" href="#id11">Multiplexed library</a><a class="headerlink" href="#multiplexed-library" title="Permalink to this headline"></a></h3>
<p>Working with paired-end raw multiplexed data.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">1.</span> <span class="n">Select</span> <span class="n">working</span> <span class="n">directory</span> <span class="n">by</span> <span class="n">pressing</span> <span class="n">the</span> <span class="s1">&#39;SELECT WORKDIR&#39;</span> <span class="n">button</span><span class="o">.</span>
</pre></div>
</div>
<div class="line-block">
<div class="line">Secify</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">sequencing</span> <span class="pre">data</span> <span class="pre">format</span></code> as <strong>multiplexed</strong>;</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">sequence</span> <span class="pre">files</span> <span class="pre">extension</span></code> as <strong>may be fastq or fasta</strong> formatted files;</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">sequencing</span> <span class="pre">read</span> <span class="pre">types</span></code> as <strong>paired-end</strong>.</div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">2.</span> <span class="s1">&#39;DEMULTIPLEX&#39;</span>
</pre></div>
</div>
<p>2.1 Press <code class="docutils literal notranslate"><span class="pre">ADD</span> <span class="pre">STEP</span></code> -&gt; <code class="docutils literal notranslate"><span class="pre">DEMULTIPLEX</span></code></p>
<p><a class="reference internal" href="_images/add_setp_demux.png"><img alt="Alternative text" src="_images/add_setp_demux.png" style="width: 300px;" /></a></p>
<p>or</p>
<p>2.2. Select <code class="docutils literal notranslate"><span class="pre">ASVs</span> <span class="pre">workflow</span></code> or <code class="docutils literal notranslate"><span class="pre">OTUs</span> <span class="pre">workflow</span></code> panel</p>
<ul class="simple">
<li><p>tick <code class="docutils literal notranslate"><span class="pre">DEMULTIPLEX</span></code>, <code class="docutils literal notranslate"><span class="pre">REORIENT</span></code> and <code class="docutils literal notranslate"><span class="pre">CUT</span> <span class="pre">PRIMERS</span></code>;</p></li>
<li><p>check that the docker is running (green icon [red = not running])</p></li>
</ul>
<p><em>(click on the image for enlargement)</em>
<a class="reference internal" href="_images/ASVs_demux_tut1.png"><img alt="Alternative text" src="_images/ASVs_demux_tut1.png" style="width: 550px;" /></a></p>
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">3.</span> <span class="n">Click</span> <span class="n">on</span> <span class="s1">&#39;DEMULTIPLEX&#39;</span> <span class="n">to</span> <span class="n">expand</span> <span class="n">the</span> <span class="n">panel</span>
</pre></div>
</div>
<ul class="simple">
<li><p>select your FASTA formatted <strong>index_file.fasta</strong> (<a class="reference internal" href="quicktools.html#indexes"><span class="std std-ref">general index file guide here</span></a>)</p></li>
<li><p>adjust <code class="docutils literal notranslate"><span class="pre">overlap</span></code> setting to fully match the length (in base pairs) of the indexes in the index_file.fasta.</p></li>
</ul>
<p><em>(click on the image for enlargement)</em>
<a class="reference internal" href="_images/demux_expand.png"><img alt="Alternative text" src="_images/demux_expand.png" style="width: 550px;" /></a></p>
<div class="line-block">
<div class="line">This step distributes sequences to samples according to the information in the index_file.fasta. See <a class="reference internal" href="quicktools.html#demux-settings"><span class="std std-ref">specifics here</span></a></div>
<div class="line"><br /></div>
<div class="line"><strong>Output</strong> directory = <code class="docutils literal notranslate"><span class="pre">demultiplex_out</span></code>:</div>
<div class="line">* fastq or fasta files per sample (as specified in the <a class="reference internal" href="quicktools.html#indexes"><span class="std std-ref">index file</span></a>)</div>
<div class="line">* unknown.fastq/fasta files contain sequences where specified index combinations were not found.</div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">1.</span>  <span class="s1">&#39;REORIENT&#39;</span>
</pre></div>
</div>
<ul class="simple">
<li><p>specify allowed <code class="docutils literal notranslate"><span class="pre">mismatches</span></code> during the primer search; &gt;2 not recommended.</p></li>
<li><p>specify <code class="docutils literal notranslate"><span class="pre">forward</span> <span class="pre">primer</span></code>: 5’-GTGYCAGCMGCCGCGGTAA-3’ (example)</p></li>
<li><p>specify <code class="docutils literal notranslate"><span class="pre">reverse</span> <span class="pre">primer</span></code>: 3’-GGCCGYCAATTYMTTTRAGTTT-5’ (example)</p></li>
</ul>
<p><em>(click on the image for enlargement)</em>
<a class="reference internal" href="_images/reorient_expand.png"><img alt="Alternative text" src="_images/reorient_expand.png" style="width: 550px;" /></a></p>
<div class="line-block">
<div class="line"><em>This step reorients sequences to 5’-3’ as based on specified forward and reverse primers. See</em> <a class="reference internal" href="quicktools.html#reorient"><span class="std std-ref">specifics here</span></a></div>
<div class="line"><br /></div>
<div class="line"><strong>Output</strong> directory = <code class="docutils literal notranslate"><span class="pre">reorient_out</span></code></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">5.</span> <span class="n">Click</span> <span class="n">on</span> <span class="s1">&#39;CUT PRIMERS&#39;</span> <span class="n">to</span> <span class="n">expand</span> <span class="n">the</span> <span class="n">panel</span>
</pre></div>
</div>
<ul class="simple">
<li><p>specify <code class="docutils literal notranslate"><span class="pre">forward</span> <span class="pre">primer</span></code>: 5’-GTGYCAGCMGCCGCGGTAA-3’ (example)</p></li>
<li><p>specify <code class="docutils literal notranslate"><span class="pre">reverse</span> <span class="pre">primer</span></code>: 3’-GGCCGYCAATTYMTTTRAGTTT-5’ (example)</p></li>
<li><p>specify allowed <code class="docutils literal notranslate"><span class="pre">mismatches</span></code> during the primer search; &gt;2 not recommended</p></li>
<li><p>for paired-end reads keep <code class="docutils literal notranslate"><span class="pre">seqs</span> <span class="pre">to</span> <span class="pre">keep</span></code> and <code class="docutils literal notranslate"><span class="pre">pair</span> <span class="pre">filter</span></code> as default (<strong>keep_all</strong> and <strong>both</strong>, respectively)</p></li>
</ul>
<p><em>(click on the image for enlargement)</em>
<a class="reference internal" href="_images/cut_primers_expand.png"><img alt="Alternative text" src="_images/cut_primers_expand.png" style="width: 550px;" /></a></p>
<div class="line-block">
<div class="line"><em>This step clipps specified primer sequences from the reads (if primers are found). See</em> <a class="reference internal" href="quicktools.html#remove-primers"><span class="std std-ref">specifics here</span></a>.</div>
<div class="line"><em>Discards the reads where primer sequences are not detected.</em></div>
<div class="line"><br /></div>
<div class="line"><strong>Output</strong> directory = <code class="docutils literal notranslate"><span class="pre">primersCut_out</span></code></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p><strong>6.</strong> Follow the rest of the <a class="reference internal" href="#rest-of-pe-asv-workflow"><span class="std std-ref">ASV workflow</span></a> or <a class="reference internal" href="#rest-of-pe-otu-workflow"><span class="std std-ref">OTU workflow</span></a></p>
</section>
</section>
<hr class="docutils" />
<section id="single-end-pacbio-or-assembled-paired-end-data">
<h2><a class="toc-backref" href="#id12">Single-end (PacBio or assembled paired-end) data</a><a class="headerlink" href="#single-end-pacbio-or-assembled-paired-end-data" title="Permalink to this headline"></a></h2>
<p>coming soon …</p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="postprocessing.html" class="btn btn-neutral float-left" title="Post-processing tools" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="troubleshoot.html" class="btn btn-neutral float-right" title="Troubleshooting" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Sten Anslan.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>